// @adyen/api-library/lib/src/typings/platformsNotificationConfiguration/testNotificationConfigurationRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TestNotificationConfigurationRequest = void 0;
class TestNotificationConfigurationRequest {
    static getAttributeTypeMap() {
        return TestNotificationConfigurationRequest.attributeTypeMap;
    }
}
exports.TestNotificationConfigurationRequest = TestNotificationConfigurationRequest;
TestNotificationConfigurationRequest.discriminator = undefined;
TestNotificationConfigurationRequest.attributeTypeMap = [
    {
        "name": "eventTypes",
        "baseName": "eventTypes",
        "type": "Array<TestNotificationConfigurationRequest.EventTypesEnum>"
    },
    {
        "name": "notificationId",
        "baseName": "notificationId",
        "type": "number"
    }
];
(function (TestNotificationConfigurationRequest) {
    let EventTypesEnum;
    (function (EventTypesEnum) {
        EventTypesEnum[EventTypesEnum["AccountClosed"] = 'ACCOUNT_CLOSED'] = "AccountClosed";
        EventTypesEnum[EventTypesEnum["AccountCreated"] = 'ACCOUNT_CREATED'] = "AccountCreated";
        EventTypesEnum[EventTypesEnum["AccountFundsBelowThreshold"] = 'ACCOUNT_FUNDS_BELOW_THRESHOLD'] = "AccountFundsBelowThreshold";
        EventTypesEnum[EventTypesEnum["AccountHolderCreated"] = 'ACCOUNT_HOLDER_CREATED'] = "AccountHolderCreated";
        EventTypesEnum[EventTypesEnum["AccountHolderLimitReached"] = 'ACCOUNT_HOLDER_LIMIT_REACHED'] = "AccountHolderLimitReached";
        EventTypesEnum[EventTypesEnum["AccountHolderPayout"] = 'ACCOUNT_HOLDER_PAYOUT'] = "AccountHolderPayout";
        EventTypesEnum[EventTypesEnum["AccountHolderStatusChange"] = 'ACCOUNT_HOLDER_STATUS_CHANGE'] = "AccountHolderStatusChange";
        EventTypesEnum[EventTypesEnum["AccountHolderStoreStatusChange"] = 'ACCOUNT_HOLDER_STORE_STATUS_CHANGE'] = "AccountHolderStoreStatusChange";
        EventTypesEnum[EventTypesEnum["AccountHolderUpcomingDeadline"] = 'ACCOUNT_HOLDER_UPCOMING_DEADLINE'] = "AccountHolderUpcomingDeadline";
        EventTypesEnum[EventTypesEnum["AccountHolderUpdated"] = 'ACCOUNT_HOLDER_UPDATED'] = "AccountHolderUpdated";
        EventTypesEnum[EventTypesEnum["AccountHolderVerification"] = 'ACCOUNT_HOLDER_VERIFICATION'] = "AccountHolderVerification";
        EventTypesEnum[EventTypesEnum["AccountUpdated"] = 'ACCOUNT_UPDATED'] = "AccountUpdated";
        EventTypesEnum[EventTypesEnum["BeneficiarySetup"] = 'BENEFICIARY_SETUP'] = "BeneficiarySetup";
        EventTypesEnum[EventTypesEnum["CompensateNegativeBalance"] = 'COMPENSATE_NEGATIVE_BALANCE'] = "CompensateNegativeBalance";
        EventTypesEnum[EventTypesEnum["DirectDebitInitiated"] = 'DIRECT_DEBIT_INITIATED'] = "DirectDebitInitiated";
        EventTypesEnum[EventTypesEnum["PaymentFailure"] = 'PAYMENT_FAILURE'] = "PaymentFailure";
        EventTypesEnum[EventTypesEnum["PendingCredit"] = 'PENDING_CREDIT'] = "PendingCredit";
        EventTypesEnum[EventTypesEnum["RefundFundsTransfer"] = 'REFUND_FUNDS_TRANSFER'] = "RefundFundsTransfer";
        EventTypesEnum[EventTypesEnum["ReportAvailable"] = 'REPORT_AVAILABLE'] = "ReportAvailable";
        EventTypesEnum[EventTypesEnum["ScheduledRefunds"] = 'SCHEDULED_REFUNDS'] = "ScheduledRefunds";
        EventTypesEnum[EventTypesEnum["ScoreSignalTriggered"] = 'SCORE_SIGNAL_TRIGGERED'] = "ScoreSignalTriggered";
        EventTypesEnum[EventTypesEnum["TransferFunds"] = 'TRANSFER_FUNDS'] = "TransferFunds";
        EventTypesEnum[EventTypesEnum["TransferNotPaidoutTransfers"] = 'TRANSFER_NOT_PAIDOUT_TRANSFERS'] = "TransferNotPaidoutTransfers";
    })(EventTypesEnum = TestNotificationConfigurationRequest.EventTypesEnum || (TestNotificationConfigurationRequest.EventTypesEnum = {}));
})(TestNotificationConfigurationRequest = exports.TestNotificationConfigurationRequest || (exports.TestNotificationConfigurationRequest = {}));
