// @adyen/api-library/lib/src/typings/payments/deviceRenderOptions.js
"use strict";
/*
 * The version of the OpenAPI document: v68
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeviceRenderOptions = void 0;
class DeviceRenderOptions {
    static getAttributeTypeMap() {
        return DeviceRenderOptions.attributeTypeMap;
    }
}
exports.DeviceRenderOptions = DeviceRenderOptions;
DeviceRenderOptions.discriminator = undefined;
DeviceRenderOptions.attributeTypeMap = [
    {
        "name": "sdkInterface",
        "baseName": "sdkInterface",
        "type": "DeviceRenderOptions.SdkInterfaceEnum"
    },
    {
        "name": "sdkUiType",
        "baseName": "sdkUiType",
        "type": "Array<DeviceRenderOptions.SdkUiTypeEnum>"
    }
];
(function (DeviceRenderOptions) {
    let SdkInterfaceEnum;
    (function (SdkInterfaceEnum) {
        SdkInterfaceEnum[SdkInterfaceEnum["Native"] = 'native'] = "Native";
        SdkInterfaceEnum[SdkInterfaceEnum["Html"] = 'html'] = "Html";
        SdkInterfaceEnum[SdkInterfaceEnum["Both"] = 'both'] = "Both";
    })(SdkInterfaceEnum = DeviceRenderOptions.SdkInterfaceEnum || (DeviceRenderOptions.SdkInterfaceEnum = {}));
    let SdkUiTypeEnum;
    (function (SdkUiTypeEnum) {
        SdkUiTypeEnum[SdkUiTypeEnum["MultiSelect"] = 'multiSelect'] = "MultiSelect";
        SdkUiTypeEnum[SdkUiTypeEnum["OtherHtml"] = 'otherHtml'] = "OtherHtml";
        SdkUiTypeEnum[SdkUiTypeEnum["OutOfBand"] = 'outOfBand'] = "OutOfBand";
        SdkUiTypeEnum[SdkUiTypeEnum["SingleSelect"] = 'singleSelect'] = "SingleSelect";
        SdkUiTypeEnum[SdkUiTypeEnum["Text"] = 'text'] = "Text";
    })(SdkUiTypeEnum = DeviceRenderOptions.SdkUiTypeEnum || (DeviceRenderOptions.SdkUiTypeEnum = {}));
})(DeviceRenderOptions = exports.DeviceRenderOptions || (exports.DeviceRenderOptions = {}));
