// @adyen/api-library/lib/src/typings/payments/modificationResult.js
"use strict";
/*
 * The version of the OpenAPI document: v68
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ModificationResult = void 0;
class ModificationResult {
    static getAttributeTypeMap() {
        return ModificationResult.attributeTypeMap;
    }
}
exports.ModificationResult = ModificationResult;
ModificationResult.discriminator = undefined;
ModificationResult.attributeTypeMap = [
    {
        "name": "additionalData",
        "baseName": "additionalData",
        "type": "{ [key: string]: string; }"
    },
    {
        "name": "pspReference",
        "baseName": "pspReference",
        "type": "string"
    },
    {
        "name": "response",
        "baseName": "response",
        "type": "ModificationResult.ResponseEnum"
    }
];
(function (ModificationResult) {
    let ResponseEnum;
    (function (ResponseEnum) {
        ResponseEnum[ResponseEnum["CaptureReceived"] = '[capture-received]'] = "CaptureReceived";
        ResponseEnum[ResponseEnum["CancelReceived"] = '[cancel-received]'] = "CancelReceived";
        ResponseEnum[ResponseEnum["RefundReceived"] = '[refund-received]'] = "RefundReceived";
        ResponseEnum[ResponseEnum["CancelOrRefundReceived"] = '[cancelOrRefund-received]'] = "CancelOrRefundReceived";
        ResponseEnum[ResponseEnum["AdjustAuthorisationReceived"] = '[adjustAuthorisation-received]'] = "AdjustAuthorisationReceived";
        ResponseEnum[ResponseEnum["DonationReceived"] = '[donation-received]'] = "DonationReceived";
        ResponseEnum[ResponseEnum["TechnicalCancelReceived"] = '[technical-cancel-received]'] = "TechnicalCancelReceived";
        ResponseEnum[ResponseEnum["VoidPendingRefundReceived"] = '[voidPendingRefund-received]'] = "VoidPendingRefundReceived";
    })(ResponseEnum = ModificationResult.ResponseEnum || (ModificationResult.ResponseEnum = {}));
})(ModificationResult = exports.ModificationResult || (exports.ModificationResult = {}));
