// @adyen/api-library/lib/src/typings/payments/threeDSecureData.js
"use strict";
/*
 * The version of the OpenAPI document: v68
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ThreeDSecureData = void 0;
class ThreeDSecureData {
    static getAttributeTypeMap() {
        return ThreeDSecureData.attributeTypeMap;
    }
}
exports.ThreeDSecureData = ThreeDSecureData;
ThreeDSecureData.discriminator = undefined;
ThreeDSecureData.attributeTypeMap = [
    {
        "name": "authenticationResponse",
        "baseName": "authenticationResponse",
        "type": "ThreeDSecureData.AuthenticationResponseEnum"
    },
    {
        "name": "cavv",
        "baseName": "cavv",
        "type": "string"
    },
    {
        "name": "cavvAlgorithm",
        "baseName": "cavvAlgorithm",
        "type": "string"
    },
    {
        "name": "challengeCancel",
        "baseName": "challengeCancel",
        "type": "ThreeDSecureData.ChallengeCancelEnum"
    },
    {
        "name": "directoryResponse",
        "baseName": "directoryResponse",
        "type": "ThreeDSecureData.DirectoryResponseEnum"
    },
    {
        "name": "dsTransID",
        "baseName": "dsTransID",
        "type": "string"
    },
    {
        "name": "eci",
        "baseName": "eci",
        "type": "string"
    },
    {
        "name": "riskScore",
        "baseName": "riskScore",
        "type": "string"
    },
    {
        "name": "threeDSVersion",
        "baseName": "threeDSVersion",
        "type": "string"
    },
    {
        "name": "tokenAuthenticationVerificationValue",
        "baseName": "tokenAuthenticationVerificationValue",
        "type": "string"
    },
    {
        "name": "transStatusReason",
        "baseName": "transStatusReason",
        "type": "string"
    },
    {
        "name": "xid",
        "baseName": "xid",
        "type": "string"
    }
];
(function (ThreeDSecureData) {
    let AuthenticationResponseEnum;
    (function (AuthenticationResponseEnum) {
        AuthenticationResponseEnum[AuthenticationResponseEnum["Y"] = 'Y'] = "Y";
        AuthenticationResponseEnum[AuthenticationResponseEnum["N"] = 'N'] = "N";
        AuthenticationResponseEnum[AuthenticationResponseEnum["U"] = 'U'] = "U";
        AuthenticationResponseEnum[AuthenticationResponseEnum["A"] = 'A'] = "A";
    })(AuthenticationResponseEnum = ThreeDSecureData.AuthenticationResponseEnum || (ThreeDSecureData.AuthenticationResponseEnum = {}));
    let ChallengeCancelEnum;
    (function (ChallengeCancelEnum) {
        ChallengeCancelEnum[ChallengeCancelEnum["_01"] = '01'] = "_01";
        ChallengeCancelEnum[ChallengeCancelEnum["_02"] = '02'] = "_02";
        ChallengeCancelEnum[ChallengeCancelEnum["_03"] = '03'] = "_03";
        ChallengeCancelEnum[ChallengeCancelEnum["_04"] = '04'] = "_04";
        ChallengeCancelEnum[ChallengeCancelEnum["_05"] = '05'] = "_05";
        ChallengeCancelEnum[ChallengeCancelEnum["_06"] = '06'] = "_06";
        ChallengeCancelEnum[ChallengeCancelEnum["_07"] = '07'] = "_07";
    })(ChallengeCancelEnum = ThreeDSecureData.ChallengeCancelEnum || (ThreeDSecureData.ChallengeCancelEnum = {}));
    let DirectoryResponseEnum;
    (function (DirectoryResponseEnum) {
        DirectoryResponseEnum[DirectoryResponseEnum["A"] = 'A'] = "A";
        DirectoryResponseEnum[DirectoryResponseEnum["C"] = 'C'] = "C";
        DirectoryResponseEnum[DirectoryResponseEnum["D"] = 'D'] = "D";
        DirectoryResponseEnum[DirectoryResponseEnum["I"] = 'I'] = "I";
        DirectoryResponseEnum[DirectoryResponseEnum["N"] = 'N'] = "N";
        DirectoryResponseEnum[DirectoryResponseEnum["R"] = 'R'] = "R";
        DirectoryResponseEnum[DirectoryResponseEnum["U"] = 'U'] = "U";
        DirectoryResponseEnum[DirectoryResponseEnum["Y"] = 'Y'] = "Y";
    })(DirectoryResponseEnum = ThreeDSecureData.DirectoryResponseEnum || (ThreeDSecureData.DirectoryResponseEnum = {}));
})(ThreeDSecureData = exports.ThreeDSecureData || (exports.ThreeDSecureData = {}));
