// @adyen/api-library/lib/src/typings/storedValue/storedValueStatusChangeRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v46
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StoredValueStatusChangeRequest = void 0;
class StoredValueStatusChangeRequest {
    static getAttributeTypeMap() {
        return StoredValueStatusChangeRequest.attributeTypeMap;
    }
}
exports.StoredValueStatusChangeRequest = StoredValueStatusChangeRequest;
StoredValueStatusChangeRequest.discriminator = undefined;
StoredValueStatusChangeRequest.attributeTypeMap = [
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount"
    },
    {
        "name": "merchantAccount",
        "baseName": "merchantAccount",
        "type": "string"
    },
    {
        "name": "paymentMethod",
        "baseName": "paymentMethod",
        "type": "{ [key: string]: string; }"
    },
    {
        "name": "recurringDetailReference",
        "baseName": "recurringDetailReference",
        "type": "string"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "shopperInteraction",
        "baseName": "shopperInteraction",
        "type": "StoredValueStatusChangeRequest.ShopperInteractionEnum"
    },
    {
        "name": "shopperReference",
        "baseName": "shopperReference",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "StoredValueStatusChangeRequest.StatusEnum"
    },
    {
        "name": "store",
        "baseName": "store",
        "type": "string"
    }
];
(function (StoredValueStatusChangeRequest) {
    let ShopperInteractionEnum;
    (function (ShopperInteractionEnum) {
        ShopperInteractionEnum[ShopperInteractionEnum["Ecommerce"] = 'Ecommerce'] = "Ecommerce";
        ShopperInteractionEnum[ShopperInteractionEnum["ContAuth"] = 'ContAuth'] = "ContAuth";
        ShopperInteractionEnum[ShopperInteractionEnum["Moto"] = 'Moto'] = "Moto";
        ShopperInteractionEnum[ShopperInteractionEnum["Pos"] = 'POS'] = "Pos";
    })(ShopperInteractionEnum = StoredValueStatusChangeRequest.ShopperInteractionEnum || (StoredValueStatusChangeRequest.ShopperInteractionEnum = {}));
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["Active"] = 'active'] = "Active";
        StatusEnum[StatusEnum["Inactive"] = 'inactive'] = "Inactive";
    })(StatusEnum = StoredValueStatusChangeRequest.StatusEnum || (StoredValueStatusChangeRequest.StatusEnum = {}));
})(StoredValueStatusChangeRequest = exports.StoredValueStatusChangeRequest || (exports.StoredValueStatusChangeRequest = {}));
