// @adyen/api-library/lib/src/services/checkout.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const apiKeyAuthenticatedService_1 = __importDefault(require("../apiKeyAuthenticatedService"));
const getJsonResponse_1 = __importDefault(require("../helpers/getJsonResponse"));
const paymentMethods_1 = __importDefault(require("./resource/checkout/paymentMethods"));
const payments_1 = __importDefault(require("./resource/checkout/payments"));
const paymentsDetails_1 = __importDefault(require("./resource/checkout/paymentsDetails"));
const paymentSession_1 = __importDefault(require("./resource/checkout/paymentSession"));
const paymentsResult_1 = __importDefault(require("./resource/checkout/paymentsResult"));
const paymentLinks_1 = __importDefault(require("./resource/checkout/paymentLinks"));
const originKeys_1 = __importDefault(require("./resource/checkout/originKeys"));
const setApplicationInfo_1 = __importDefault(require("../helpers/setApplicationInfo"));
const amountUpdates_1 = __importDefault(require("./resource/checkout/amountUpdates"));
const cancels_1 = __importDefault(require("./resource/checkout/cancels"));
const captures_1 = __importDefault(require("./resource/checkout/captures"));
const refunds_1 = __importDefault(require("./resource/checkout/refunds"));
const reversals_1 = __importDefault(require("./resource/checkout/reversals"));
const cancelsStandalone_1 = __importDefault(require("./resource/checkout/cancelsStandalone"));
const models_1 = require("../typings/checkout/models");
const paymentLinksId_1 = __importDefault(require("./resource/checkout/paymentLinksId"));
const paymentMethodsBalance_1 = __importDefault(require("./resource/checkout/paymentMethodsBalance"));
const orders_1 = __importDefault(require("./resource/checkout/orders"));
const ordersCancel_1 = __importDefault(require("./resource/checkout/ordersCancel"));
const sessions_1 = __importDefault(require("./resource/checkout/sessions"));
const donations_1 = __importDefault(require("./resource/checkout/donations"));
const cardDetails_1 = __importDefault(require("./resource/checkout/cardDetails"));
class Checkout extends apiKeyAuthenticatedService_1.default {
    constructor(client) {
        super(client);
        this._payments = new payments_1.default(this);
        this._paymentMethods = new paymentMethods_1.default(this);
        this._paymentsDetails = new paymentsDetails_1.default(this);
        this._paymentSession = new paymentSession_1.default(this);
        this._paymentsResult = new paymentsResult_1.default(this);
        this._paymentLinks = new paymentLinks_1.default(this);
        this._originKeys = new originKeys_1.default(this);
        this._paymentMethodsBalance = new paymentMethodsBalance_1.default(this);
        this._orders = new orders_1.default(this);
        this._ordersCancel = new ordersCancel_1.default(this);
        this._sessions = new sessions_1.default(this);
        this._donations = new donations_1.default(this);
        this._cardDetails = new cardDetails_1.default(this);
    }
    // Payments
    async sessions(checkoutSessionRequest, requestOptions) {
        const response = await (0, getJsonResponse_1.default)(this._sessions, checkoutSessionRequest, requestOptions);
        return models_1.ObjectSerializer.deserialize(response, "CreateCheckoutSessionResponse");
    }
    async paymentMethods(paymentMethodsRequest, requestOptions) {
        const response = await (0, getJsonResponse_1.default)(this._paymentMethods, paymentMethodsRequest, requestOptions);
        return models_1.ObjectSerializer.deserialize(response, "PaymentMethodsResponse");
    }
    async payments(paymentsRequest, requestOptions) {
        const response = await (0, getJsonResponse_1.default)(this._payments, (0, setApplicationInfo_1.default)(paymentsRequest), requestOptions);
        return models_1.ObjectSerializer.deserialize(response, "PaymentResponse");
    }
    async paymentsDetails(paymentsDetailsRequest, requestOptions) {
        const response = await (0, getJsonResponse_1.default)(this._paymentsDetails, paymentsDetailsRequest, requestOptions);
        return models_1.ObjectSerializer.deserialize(response, "PaymentResponse");
    }
    async donations(donationRequest, requestOptions) {
        const response = await (0, getJsonResponse_1.default)(this._donations, donationRequest, requestOptions);
        return models_1.ObjectSerializer.deserialize(response, "DonationResponse");
    }
    async cardDetails(cardDetailsRequest, requestOptions) {
        const response = await (0, getJsonResponse_1.default)(this._cardDetails, cardDetailsRequest, requestOptions);
        return models_1.ObjectSerializer.deserialize(response, "CardDetailsResponse");
    }
    // Payment Links
    async paymentLinks(paymentLinkRequest, requestOptions) {
        const response = await (0, getJsonResponse_1.default)(this._paymentLinks, paymentLinkRequest, requestOptions);
        return models_1.ObjectSerializer.deserialize(response, "PaymentLinkResponse");
    }
    async getPaymentLinks(linkId, requestOptions) {
        const paymentLinksId = new paymentLinksId_1.default(this, linkId);
        const response = await (0, getJsonResponse_1.default)(paymentLinksId, {}, { ...{ method: "GET" }, ...requestOptions });
        return models_1.ObjectSerializer.deserialize(response, "PaymentLinkResponse");
    }
    async updatePaymentLinks(linkId, status, requestOptions) {
        const paymentLinksId = new paymentLinksId_1.default(this, linkId);
        const response = await (0, getJsonResponse_1.default)(paymentLinksId, { status }, { ...{ method: "PATCH" }, ...requestOptions });
        return models_1.ObjectSerializer.deserialize(response, "PaymentLinkResponse");
    }
    // Modifications
    async amountUpdates(paymentPspReference, amountUpdatesRequest, requestOptions) {
        const amountUpdates = new amountUpdates_1.default(this, paymentPspReference);
        const response = await (0, getJsonResponse_1.default)(amountUpdates, amountUpdatesRequest, requestOptions);
        return models_1.ObjectSerializer.deserialize(response, "PaymentAmountUpdateResource");
    }
    async cancelsStandalone(cancelsStandaloneRequest, requestOptions) {
        const cancelsStandalone = new cancelsStandalone_1.default(this);
        const response = await (0, getJsonResponse_1.default)(cancelsStandalone, cancelsStandaloneRequest, requestOptions);
        return models_1.ObjectSerializer.deserialize(response, "StandalonePaymentCancelResource");
    }
    async cancels(paymentPspReference, cancelsRequest, requestOptions) {
        const cancels = new cancels_1.default(this, paymentPspReference);
        const response = await (0, getJsonResponse_1.default)(cancels, cancelsRequest, requestOptions);
        return models_1.ObjectSerializer.deserialize(response, "PaymentCancelResource");
    }
    async captures(paymentPspReference, capturesRequest, requestOptions) {
        const captures = new captures_1.default(this, paymentPspReference);
        const response = await (0, getJsonResponse_1.default)(captures, capturesRequest, requestOptions);
        return models_1.ObjectSerializer.deserialize(response, "PaymentCaptureResource");
    }
    async refunds(paymentPspReference, refundsRequest, requestOptions) {
        const refunds = new refunds_1.default(this, paymentPspReference);
        const response = await (0, getJsonResponse_1.default)(refunds, refundsRequest, requestOptions);
        return models_1.ObjectSerializer.deserialize(response, "PaymentRefundResource");
    }
    async reversals(paymentPspReference, reversalsRequest, requestOptions) {
        const refunds = new reversals_1.default(this, paymentPspReference);
        const response = await (0, getJsonResponse_1.default)(refunds, reversalsRequest, requestOptions);
        return models_1.ObjectSerializer.deserialize(response, "PaymentReversalResource");
    }
    // Orders
    async paymentMethodsBalance(paymentMethodsBalanceRequest, requestOptions) {
        const response = await (0, getJsonResponse_1.default)(this._paymentMethodsBalance, paymentMethodsBalanceRequest, requestOptions);
        return models_1.ObjectSerializer.deserialize(response, "CheckoutBalanceCheckResponse");
    }
    async orders(ordersRequest, requestOptions) {
        const response = await (0, getJsonResponse_1.default)(this._orders, ordersRequest, requestOptions);
        return models_1.ObjectSerializer.deserialize(response, "CheckoutCreateOrderResponse");
    }
    async ordersCancel(ordersCancelRequest, requestOptions) {
        const response = await (0, getJsonResponse_1.default)(this._ordersCancel, ordersCancelRequest, requestOptions);
        return models_1.ObjectSerializer.deserialize(response, "CheckoutCancelOrderResponse");
    }
    // Classic Checkout SDK
    async paymentSession(paymentSessionRequest, requestOptions) {
        const response = await (0, getJsonResponse_1.default)(this._paymentSession, paymentSessionRequest, requestOptions);
        return models_1.ObjectSerializer.deserialize(response, "PaymentSetupResponse");
    }
    async paymentResult(paymentResultRequest, requestOptions) {
        const response = await (0, getJsonResponse_1.default)(this._paymentsResult, paymentResultRequest, requestOptions);
        return models_1.ObjectSerializer.deserialize(response, "PaymentVerificationResponse");
    }
    //Utility
    async originKeys(originKeysRequest, requestOptions) {
        const response = await (0, getJsonResponse_1.default)(this._originKeys, originKeysRequest, requestOptions);
        return models_1.ObjectSerializer.deserialize(response, "CheckoutUtilityResponse");
    }
}
exports.default = Checkout;
