// @adyen/api-library/lib/src/typings/checkout/afterpayDetails.js
"use strict";
/*
 * The version of the OpenAPI document: v69
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AfterpayDetails = void 0;
class AfterpayDetails {
    static getAttributeTypeMap() {
        return AfterpayDetails.attributeTypeMap;
    }
}
exports.AfterpayDetails = AfterpayDetails;
AfterpayDetails.discriminator = undefined;
AfterpayDetails.attributeTypeMap = [
    {
        "name": "billingAddress",
        "baseName": "billingAddress",
        "type": "string"
    },
    {
        "name": "deliveryAddress",
        "baseName": "deliveryAddress",
        "type": "string"
    },
    {
        "name": "personalDetails",
        "baseName": "personalDetails",
        "type": "string"
    },
    {
        "name": "recurringDetailReference",
        "baseName": "recurringDetailReference",
        "type": "string"
    },
    {
        "name": "storedPaymentMethodId",
        "baseName": "storedPaymentMethodId",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "AfterpayDetails.TypeEnum"
    }
];
(function (AfterpayDetails) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["AfterpayDefault"] = 'afterpay_default'] = "AfterpayDefault";
        TypeEnum[TypeEnum["Afterpaytouch"] = 'afterpaytouch'] = "Afterpaytouch";
        TypeEnum[TypeEnum["AfterpayB2b"] = 'afterpay_b2b'] = "AfterpayB2b";
    })(TypeEnum = AfterpayDetails.TypeEnum || (AfterpayDetails.TypeEnum = {}));
})(AfterpayDetails = exports.AfterpayDetails || (exports.AfterpayDetails = {}));
