// @adyen/api-library/lib/src/typings/checkout/billDeskDetails.js
"use strict";
/*
 * The version of the OpenAPI document: v69
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BillDeskDetails = void 0;
class BillDeskDetails {
    static getAttributeTypeMap() {
        return BillDeskDetails.attributeTypeMap;
    }
}
exports.BillDeskDetails = BillDeskDetails;
BillDeskDetails.discriminator = undefined;
BillDeskDetails.attributeTypeMap = [
    {
        "name": "issuer",
        "baseName": "issuer",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "BillDeskDetails.TypeEnum"
    }
];
(function (BillDeskDetails) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["BilldeskOnline"] = 'billdesk_online'] = "BilldeskOnline";
        TypeEnum[TypeEnum["BilldeskWallet"] = 'billdesk_wallet'] = "BilldeskWallet";
        TypeEnum[TypeEnum["OnlinebankingIn"] = 'onlinebanking_IN'] = "OnlinebankingIn";
        TypeEnum[TypeEnum["WalletIn"] = 'wallet_IN'] = "WalletIn";
    })(TypeEnum = BillDeskDetails.TypeEnum || (BillDeskDetails.TypeEnum = {}));
})(BillDeskDetails = exports.BillDeskDetails || (exports.BillDeskDetails = {}));
