// @adyen/api-library/lib/src/typings/checkout/dokuDetails.js
"use strict";
/*
 * The version of the OpenAPI document: v69
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DokuDetails = void 0;
class DokuDetails {
    static getAttributeTypeMap() {
        return DokuDetails.attributeTypeMap;
    }
}
exports.DokuDetails = DokuDetails;
DokuDetails.discriminator = undefined;
DokuDetails.attributeTypeMap = [
    {
        "name": "firstName",
        "baseName": "firstName",
        "type": "string"
    },
    {
        "name": "lastName",
        "baseName": "lastName",
        "type": "string"
    },
    {
        "name": "shopperEmail",
        "baseName": "shopperEmail",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "DokuDetails.TypeEnum"
    }
];
(function (DokuDetails) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["MandiriVa"] = 'doku_mandiri_va'] = "MandiriVa";
        TypeEnum[TypeEnum["CimbVa"] = 'doku_cimb_va'] = "CimbVa";
        TypeEnum[TypeEnum["DanamonVa"] = 'doku_danamon_va'] = "DanamonVa";
        TypeEnum[TypeEnum["BniVa"] = 'doku_bni_va'] = "BniVa";
        TypeEnum[TypeEnum["PermataLiteAtm"] = 'doku_permata_lite_atm'] = "PermataLiteAtm";
        TypeEnum[TypeEnum["BriVa"] = 'doku_bri_va'] = "BriVa";
        TypeEnum[TypeEnum["BcaVa"] = 'doku_bca_va'] = "BcaVa";
        TypeEnum[TypeEnum["Alfamart"] = 'doku_alfamart'] = "Alfamart";
        TypeEnum[TypeEnum["Indomaret"] = 'doku_indomaret'] = "Indomaret";
    })(TypeEnum = DokuDetails.TypeEnum || (DokuDetails.TypeEnum = {}));
})(DokuDetails = exports.DokuDetails || (exports.DokuDetails = {}));
