// @adyen/api-library/lib/src/typings/checkout/dragonpayDetails.js
"use strict";
/*
 * The version of the OpenAPI document: v69
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DragonpayDetails = void 0;
class DragonpayDetails {
    static getAttributeTypeMap() {
        return DragonpayDetails.attributeTypeMap;
    }
}
exports.DragonpayDetails = DragonpayDetails;
DragonpayDetails.discriminator = undefined;
DragonpayDetails.attributeTypeMap = [
    {
        "name": "issuer",
        "baseName": "issuer",
        "type": "string"
    },
    {
        "name": "shopperEmail",
        "baseName": "shopperEmail",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "DragonpayDetails.TypeEnum"
    }
];
(function (DragonpayDetails) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["Ebanking"] = 'dragonpay_ebanking'] = "Ebanking";
        TypeEnum[TypeEnum["OtcBanking"] = 'dragonpay_otc_banking'] = "OtcBanking";
        TypeEnum[TypeEnum["OtcNonBanking"] = 'dragonpay_otc_non_banking'] = "OtcNonBanking";
        TypeEnum[TypeEnum["OtcPhilippines"] = 'dragonpay_otc_philippines'] = "OtcPhilippines";
    })(TypeEnum = DragonpayDetails.TypeEnum || (DragonpayDetails.TypeEnum = {}));
})(DragonpayDetails = exports.DragonpayDetails || (exports.DragonpayDetails = {}));
