// @adyen/api-library/lib/src/typings/checkout/genericIssuerPaymentMethodDetails.js
"use strict";
/*
 * The version of the OpenAPI document: v69
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GenericIssuerPaymentMethodDetails = void 0;
class GenericIssuerPaymentMethodDetails {
    static getAttributeTypeMap() {
        return GenericIssuerPaymentMethodDetails.attributeTypeMap;
    }
}
exports.GenericIssuerPaymentMethodDetails = GenericIssuerPaymentMethodDetails;
GenericIssuerPaymentMethodDetails.discriminator = undefined;
GenericIssuerPaymentMethodDetails.attributeTypeMap = [
    {
        "name": "issuer",
        "baseName": "issuer",
        "type": "string"
    },
    {
        "name": "recurringDetailReference",
        "baseName": "recurringDetailReference",
        "type": "string"
    },
    {
        "name": "storedPaymentMethodId",
        "baseName": "storedPaymentMethodId",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "GenericIssuerPaymentMethodDetails.TypeEnum"
    }
];
(function (GenericIssuerPaymentMethodDetails) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["Eps"] = 'eps'] = "Eps";
        TypeEnum[TypeEnum["OnlineBankingSk"] = 'onlineBanking_SK'] = "OnlineBankingSk";
        TypeEnum[TypeEnum["OnlineBankingCz"] = 'onlineBanking_CZ'] = "OnlineBankingCz";
    })(TypeEnum = GenericIssuerPaymentMethodDetails.TypeEnum || (GenericIssuerPaymentMethodDetails.TypeEnum = {}));
})(GenericIssuerPaymentMethodDetails = exports.GenericIssuerPaymentMethodDetails || (exports.GenericIssuerPaymentMethodDetails = {}));
