// @adyen/api-library/lib/src/typings/checkout/klarnaDetails.js
"use strict";
/*
 * The version of the OpenAPI document: v69
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.KlarnaDetails = void 0;
class KlarnaDetails {
    static getAttributeTypeMap() {
        return KlarnaDetails.attributeTypeMap;
    }
}
exports.KlarnaDetails = KlarnaDetails;
KlarnaDetails.discriminator = undefined;
KlarnaDetails.attributeTypeMap = [
    {
        "name": "billingAddress",
        "baseName": "billingAddress",
        "type": "string"
    },
    {
        "name": "deliveryAddress",
        "baseName": "deliveryAddress",
        "type": "string"
    },
    {
        "name": "personalDetails",
        "baseName": "personalDetails",
        "type": "string"
    },
    {
        "name": "recurringDetailReference",
        "baseName": "recurringDetailReference",
        "type": "string"
    },
    {
        "name": "storedPaymentMethodId",
        "baseName": "storedPaymentMethodId",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "KlarnaDetails.TypeEnum"
    }
];
(function (KlarnaDetails) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["Klarna"] = 'klarna'] = "Klarna";
        TypeEnum[TypeEnum["Klarnapayments"] = 'klarnapayments'] = "Klarnapayments";
        TypeEnum[TypeEnum["KlarnapaymentsAccount"] = 'klarnapayments_account'] = "KlarnapaymentsAccount";
        TypeEnum[TypeEnum["KlarnapaymentsB2b"] = 'klarnapayments_b2b'] = "KlarnapaymentsB2b";
        TypeEnum[TypeEnum["KlarnaPaynow"] = 'klarna_paynow'] = "KlarnaPaynow";
        TypeEnum[TypeEnum["KlarnaAccount"] = 'klarna_account'] = "KlarnaAccount";
        TypeEnum[TypeEnum["KlarnaB2b"] = 'klarna_b2b'] = "KlarnaB2b";
    })(TypeEnum = KlarnaDetails.TypeEnum || (KlarnaDetails.TypeEnum = {}));
})(KlarnaDetails = exports.KlarnaDetails || (exports.KlarnaDetails = {}));
