// @adyen/api-library/lib/src/typings/checkout/molPayDetails.js
"use strict";
/*
 * The version of the OpenAPI document: v69
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MolPayDetails = void 0;
class MolPayDetails {
    static getAttributeTypeMap() {
        return MolPayDetails.attributeTypeMap;
    }
}
exports.MolPayDetails = MolPayDetails;
MolPayDetails.discriminator = undefined;
MolPayDetails.attributeTypeMap = [
    {
        "name": "issuer",
        "baseName": "issuer",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "MolPayDetails.TypeEnum"
    }
];
(function (MolPayDetails) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["FpxMy"] = 'molpay_ebanking_fpx_MY'] = "FpxMy";
        TypeEnum[TypeEnum["Th"] = 'molpay_ebanking_TH'] = "Th";
    })(TypeEnum = MolPayDetails.TypeEnum || (MolPayDetails.TypeEnum = {}));
})(MolPayDetails = exports.MolPayDetails || (exports.MolPayDetails = {}));
