// @adyen/api-library/lib/src/typings/checkout/paymentDetails.js
"use strict";
/*
 * The version of the OpenAPI document: v69
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentDetails = void 0;
class PaymentDetails {
    static getAttributeTypeMap() {
        return PaymentDetails.attributeTypeMap;
    }
}
exports.PaymentDetails = PaymentDetails;
PaymentDetails.discriminator = undefined;
PaymentDetails.attributeTypeMap = [
    {
        "name": "type",
        "baseName": "type",
        "type": "PaymentDetails.TypeEnum"
    }
];
(function (PaymentDetails) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["Alipay"] = 'alipay'] = "Alipay";
        TypeEnum[TypeEnum["Multibanco"] = 'multibanco'] = "Multibanco";
        TypeEnum[TypeEnum["BankTransferIban"] = 'bankTransfer_IBAN'] = "BankTransferIban";
        TypeEnum[TypeEnum["Paybright"] = 'paybright'] = "Paybright";
        TypeEnum[TypeEnum["Paynow"] = 'paynow'] = "Paynow";
        TypeEnum[TypeEnum["Affirm"] = 'affirm'] = "Affirm";
        TypeEnum[TypeEnum["AffirmPos"] = 'affirm_pos'] = "AffirmPos";
        TypeEnum[TypeEnum["Trustlyvector"] = 'trustlyvector'] = "Trustlyvector";
        TypeEnum[TypeEnum["Oney"] = 'oney'] = "Oney";
        TypeEnum[TypeEnum["Facilypay"] = 'facilypay'] = "Facilypay";
        TypeEnum[TypeEnum["Facilypay3x"] = 'facilypay_3x'] = "Facilypay3x";
        TypeEnum[TypeEnum["Facilypay4x"] = 'facilypay_4x'] = "Facilypay4x";
        TypeEnum[TypeEnum["Facilypay6x"] = 'facilypay_6x'] = "Facilypay6x";
        TypeEnum[TypeEnum["Facilypay10x"] = 'facilypay_10x'] = "Facilypay10x";
        TypeEnum[TypeEnum["Facilypay12x"] = 'facilypay_12x'] = "Facilypay12x";
        TypeEnum[TypeEnum["Unionpay"] = 'unionpay'] = "Unionpay";
        TypeEnum[TypeEnum["KcpBanktransfer"] = 'kcp_banktransfer'] = "KcpBanktransfer";
        TypeEnum[TypeEnum["KcpPayco"] = 'kcp_payco'] = "KcpPayco";
        TypeEnum[TypeEnum["KcpCreditcard"] = 'kcp_creditcard'] = "KcpCreditcard";
        TypeEnum[TypeEnum["WechatpaySdk"] = 'wechatpaySDK'] = "WechatpaySdk";
        TypeEnum[TypeEnum["WechatpayQr"] = 'wechatpayQR'] = "WechatpayQr";
        TypeEnum[TypeEnum["WechatpayWeb"] = 'wechatpayWeb'] = "WechatpayWeb";
        TypeEnum[TypeEnum["MolpayBoost"] = 'molpay_boost'] = "MolpayBoost";
        TypeEnum[TypeEnum["WalletIn"] = 'wallet_IN'] = "WalletIn";
        TypeEnum[TypeEnum["PayuInCashcard"] = 'payu_IN_cashcard'] = "PayuInCashcard";
        TypeEnum[TypeEnum["PayuInNb"] = 'payu_IN_nb'] = "PayuInNb";
        TypeEnum[TypeEnum["UpiQr"] = 'upi_qr'] = "UpiQr";
        TypeEnum[TypeEnum["Paytm"] = 'paytm'] = "Paytm";
        TypeEnum[TypeEnum["MolpayEbankingVn"] = 'molpay_ebanking_VN'] = "MolpayEbankingVn";
        TypeEnum[TypeEnum["OnlineBankingPl"] = 'onlineBanking_PL'] = "OnlineBankingPl";
        TypeEnum[TypeEnum["OpenbankingUk"] = 'openbanking_UK'] = "OpenbankingUk";
        TypeEnum[TypeEnum["EbankingFi"] = 'ebanking_FI'] = "EbankingFi";
        TypeEnum[TypeEnum["MolpayEbankingMy"] = 'molpay_ebanking_MY'] = "MolpayEbankingMy";
        TypeEnum[TypeEnum["MolpayEbankingDirectMy"] = 'molpay_ebanking_direct_MY'] = "MolpayEbankingDirectMy";
        TypeEnum[TypeEnum["Swish"] = 'swish'] = "Swish";
        TypeEnum[TypeEnum["Twint"] = 'twint'] = "Twint";
        TypeEnum[TypeEnum["Pix"] = 'pix'] = "Pix";
        TypeEnum[TypeEnum["Walley"] = 'walley'] = "Walley";
        TypeEnum[TypeEnum["WalleyB2b"] = 'walley_b2b'] = "WalleyB2b";
        TypeEnum[TypeEnum["Alma"] = 'alma'] = "Alma";
        TypeEnum[TypeEnum["MolpayFpx"] = 'molpay_fpx'] = "MolpayFpx";
        TypeEnum[TypeEnum["Konbini"] = 'konbini'] = "Konbini";
        TypeEnum[TypeEnum["DirectEbanking"] = 'directEbanking'] = "DirectEbanking";
        TypeEnum[TypeEnum["Boletobancario"] = 'boletobancario'] = "Boletobancario";
        TypeEnum[TypeEnum["Neteller"] = 'neteller'] = "Neteller";
        TypeEnum[TypeEnum["Dana"] = 'dana'] = "Dana";
        TypeEnum[TypeEnum["Paysafecard"] = 'paysafecard'] = "Paysafecard";
        TypeEnum[TypeEnum["Cashticket"] = 'cashticket'] = "Cashticket";
        TypeEnum[TypeEnum["Isracard"] = 'isracard'] = "Isracard";
        TypeEnum[TypeEnum["Ikano"] = 'ikano'] = "Ikano";
        TypeEnum[TypeEnum["Karenmillen"] = 'karenmillen'] = "Karenmillen";
        TypeEnum[TypeEnum["Oasis"] = 'oasis'] = "Oasis";
        TypeEnum[TypeEnum["Warehouse"] = 'warehouse'] = "Warehouse";
        TypeEnum[TypeEnum["PrimeiropayBoleto"] = 'primeiropay_boleto'] = "PrimeiropayBoleto";
        TypeEnum[TypeEnum["Mada"] = 'mada'] = "Mada";
        TypeEnum[TypeEnum["Benefit"] = 'benefit'] = "Benefit";
        TypeEnum[TypeEnum["Knet"] = 'knet'] = "Knet";
        TypeEnum[TypeEnum["Omannet"] = 'omannet'] = "Omannet";
        TypeEnum[TypeEnum["GopayWallet"] = 'gopay_wallet'] = "GopayWallet";
        TypeEnum[TypeEnum["Poli"] = 'poli'] = "Poli";
        TypeEnum[TypeEnum["KcpNaverpay"] = 'kcp_naverpay'] = "KcpNaverpay";
        TypeEnum[TypeEnum["OnlinebankingIn"] = 'onlinebanking_IN'] = "OnlinebankingIn";
        TypeEnum[TypeEnum["Fawry"] = 'fawry'] = "Fawry";
        TypeEnum[TypeEnum["Atome"] = 'atome'] = "Atome";
        TypeEnum[TypeEnum["Moneybookers"] = 'moneybookers'] = "Moneybookers";
        TypeEnum[TypeEnum["Naps"] = 'naps'] = "Naps";
        TypeEnum[TypeEnum["Nordea"] = 'nordea'] = "Nordea";
        TypeEnum[TypeEnum["BoletobancarioBradesco"] = 'boletobancario_bradesco'] = "BoletobancarioBradesco";
        TypeEnum[TypeEnum["BoletobancarioItau"] = 'boletobancario_itau'] = "BoletobancarioItau";
        TypeEnum[TypeEnum["BoletobancarioSantander"] = 'boletobancario_santander'] = "BoletobancarioSantander";
        TypeEnum[TypeEnum["BoletobancarioBancodobrasil"] = 'boletobancario_bancodobrasil'] = "BoletobancarioBancodobrasil";
        TypeEnum[TypeEnum["BoletobancarioHsbc"] = 'boletobancario_hsbc'] = "BoletobancarioHsbc";
        TypeEnum[TypeEnum["MolpayMaybank2u"] = 'molpay_maybank2u'] = "MolpayMaybank2u";
        TypeEnum[TypeEnum["MolpayCimb"] = 'molpay_cimb'] = "MolpayCimb";
        TypeEnum[TypeEnum["MolpayRhb"] = 'molpay_rhb'] = "MolpayRhb";
        TypeEnum[TypeEnum["MolpayAmb"] = 'molpay_amb'] = "MolpayAmb";
        TypeEnum[TypeEnum["MolpayHlb"] = 'molpay_hlb'] = "MolpayHlb";
        TypeEnum[TypeEnum["MolpayAffinEpg"] = 'molpay_affin_epg'] = "MolpayAffinEpg";
        TypeEnum[TypeEnum["MolpayBankislam"] = 'molpay_bankislam'] = "MolpayBankislam";
        TypeEnum[TypeEnum["MolpayPublicbank"] = 'molpay_publicbank'] = "MolpayPublicbank";
        TypeEnum[TypeEnum["FpxAgrobank"] = 'fpx_agrobank'] = "FpxAgrobank";
        TypeEnum[TypeEnum["Touchngo"] = 'touchngo'] = "Touchngo";
        TypeEnum[TypeEnum["Maybank2uMae"] = 'maybank2u_mae'] = "Maybank2uMae";
        TypeEnum[TypeEnum["Duitnow"] = 'duitnow'] = "Duitnow";
        TypeEnum[TypeEnum["Promptpay"] = 'promptpay'] = "Promptpay";
        TypeEnum[TypeEnum["TwintPos"] = 'twint_pos'] = "TwintPos";
        TypeEnum[TypeEnum["AlipayHk"] = 'alipay_hk'] = "AlipayHk";
        TypeEnum[TypeEnum["AlipayHkWeb"] = 'alipay_hk_web'] = "AlipayHkWeb";
        TypeEnum[TypeEnum["AlipayHkWap"] = 'alipay_hk_wap'] = "AlipayHkWap";
        TypeEnum[TypeEnum["AlipayWap"] = 'alipay_wap'] = "AlipayWap";
        TypeEnum[TypeEnum["Balanceplatform"] = 'balanceplatform'] = "Balanceplatform";
    })(TypeEnum = PaymentDetails.TypeEnum || (PaymentDetails.TypeEnum = {}));
})(PaymentDetails = exports.PaymentDetails || (exports.PaymentDetails = {}));
