// @adyen/api-library/lib/src/typings/checkout/paymentAmountUpdateResource.js
"use strict";
/*
 * The version of the OpenAPI document: v69
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentAmountUpdateResource = void 0;
class PaymentAmountUpdateResource {
    static getAttributeTypeMap() {
        return PaymentAmountUpdateResource.attributeTypeMap;
    }
}
exports.PaymentAmountUpdateResource = PaymentAmountUpdateResource;
PaymentAmountUpdateResource.discriminator = undefined;
PaymentAmountUpdateResource.attributeTypeMap = [
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount"
    },
    {
        "name": "merchantAccount",
        "baseName": "merchantAccount",
        "type": "string"
    },
    {
        "name": "paymentPspReference",
        "baseName": "paymentPspReference",
        "type": "string"
    },
    {
        "name": "pspReference",
        "baseName": "pspReference",
        "type": "string"
    },
    {
        "name": "reason",
        "baseName": "reason",
        "type": "PaymentAmountUpdateResource.ReasonEnum"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "splits",
        "baseName": "splits",
        "type": "Array<Split>"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "PaymentAmountUpdateResource.StatusEnum"
    }
];
(function (PaymentAmountUpdateResource) {
    let ReasonEnum;
    (function (ReasonEnum) {
        ReasonEnum[ReasonEnum["DelayedCharge"] = 'delayedCharge'] = "DelayedCharge";
        ReasonEnum[ReasonEnum["NoShow"] = 'noShow'] = "NoShow";
    })(ReasonEnum = PaymentAmountUpdateResource.ReasonEnum || (PaymentAmountUpdateResource.ReasonEnum = {}));
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["Received"] = 'received'] = "Received";
    })(StatusEnum = PaymentAmountUpdateResource.StatusEnum || (PaymentAmountUpdateResource.StatusEnum = {}));
})(PaymentAmountUpdateResource = exports.PaymentAmountUpdateResource || (exports.PaymentAmountUpdateResource = {}));
