// @adyen/api-library/lib/src/typings/checkout/recurring.js
"use strict";
/*
 * The version of the OpenAPI document: v69
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Recurring = void 0;
class Recurring {
    static getAttributeTypeMap() {
        return Recurring.attributeTypeMap;
    }
}
exports.Recurring = Recurring;
Recurring.discriminator = undefined;
Recurring.attributeTypeMap = [
    {
        "name": "contract",
        "baseName": "contract",
        "type": "Recurring.ContractEnum"
    },
    {
        "name": "recurringDetailName",
        "baseName": "recurringDetailName",
        "type": "string"
    },
    {
        "name": "recurringExpiry",
        "baseName": "recurringExpiry",
        "type": "Date"
    },
    {
        "name": "recurringFrequency",
        "baseName": "recurringFrequency",
        "type": "string"
    },
    {
        "name": "tokenService",
        "baseName": "tokenService",
        "type": "Recurring.TokenServiceEnum"
    }
];
(function (Recurring) {
    let ContractEnum;
    (function (ContractEnum) {
        ContractEnum[ContractEnum["Oneclick"] = 'ONECLICK'] = "Oneclick";
        ContractEnum[ContractEnum["Recurring"] = 'RECURRING'] = "Recurring";
        ContractEnum[ContractEnum["Payout"] = 'PAYOUT'] = "Payout";
    })(ContractEnum = Recurring.ContractEnum || (Recurring.ContractEnum = {}));
    let TokenServiceEnum;
    (function (TokenServiceEnum) {
        TokenServiceEnum[TokenServiceEnum["Visatokenservice"] = 'VISATOKENSERVICE'] = "Visatokenservice";
        TokenServiceEnum[TokenServiceEnum["Mctokenservice"] = 'MCTOKENSERVICE'] = "Mctokenservice";
    })(TokenServiceEnum = Recurring.TokenServiceEnum || (Recurring.TokenServiceEnum = {}));
})(Recurring = exports.Recurring || (exports.Recurring = {}));
