// @adyen/api-library/lib/src/typings/checkout/redirect.js
"use strict";
/*
 * The version of the OpenAPI document: v69
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Redirect = void 0;
class Redirect {
    static getAttributeTypeMap() {
        return Redirect.attributeTypeMap;
    }
}
exports.Redirect = Redirect;
Redirect.discriminator = undefined;
Redirect.attributeTypeMap = [
    {
        "name": "data",
        "baseName": "data",
        "type": "{ [key: string]: string; }"
    },
    {
        "name": "method",
        "baseName": "method",
        "type": "Redirect.MethodEnum"
    },
    {
        "name": "url",
        "baseName": "url",
        "type": "string"
    }
];
(function (Redirect) {
    let MethodEnum;
    (function (MethodEnum) {
        MethodEnum[MethodEnum["Get"] = 'GET'] = "Get";
        MethodEnum[MethodEnum["Post"] = 'POST'] = "Post";
    })(MethodEnum = Redirect.MethodEnum || (Redirect.MethodEnum = {}));
})(Redirect = exports.Redirect || (exports.Redirect = {}));
