// @adyen/api-library/lib/src/typings/checkout/sepaDirectDebitDetails.js
"use strict";
/*
 * The version of the OpenAPI document: v69
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SepaDirectDebitDetails = void 0;
class SepaDirectDebitDetails {
    static getAttributeTypeMap() {
        return SepaDirectDebitDetails.attributeTypeMap;
    }
}
exports.SepaDirectDebitDetails = SepaDirectDebitDetails;
SepaDirectDebitDetails.discriminator = undefined;
SepaDirectDebitDetails.attributeTypeMap = [
    {
        "name": "iban",
        "baseName": "iban",
        "type": "string"
    },
    {
        "name": "ownerName",
        "baseName": "ownerName",
        "type": "string"
    },
    {
        "name": "recurringDetailReference",
        "baseName": "recurringDetailReference",
        "type": "string"
    },
    {
        "name": "storedPaymentMethodId",
        "baseName": "storedPaymentMethodId",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "SepaDirectDebitDetails.TypeEnum"
    }
];
(function (SepaDirectDebitDetails) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["Sepadirectdebit"] = 'sepadirectdebit'] = "Sepadirectdebit";
        TypeEnum[TypeEnum["SepadirectdebitAmazonpay"] = 'sepadirectdebit_amazonpay'] = "SepadirectdebitAmazonpay";
    })(TypeEnum = SepaDirectDebitDetails.TypeEnum || (SepaDirectDebitDetails.TypeEnum = {}));
})(SepaDirectDebitDetails = exports.SepaDirectDebitDetails || (exports.SepaDirectDebitDetails = {}));
