// @adyen/api-library/lib/src/typings/checkout/shopperInput.js
"use strict";
/*
 * The version of the OpenAPI document: v69
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShopperInput = void 0;
class ShopperInput {
    static getAttributeTypeMap() {
        return ShopperInput.attributeTypeMap;
    }
}
exports.ShopperInput = ShopperInput;
ShopperInput.discriminator = undefined;
ShopperInput.attributeTypeMap = [
    {
        "name": "billingAddress",
        "baseName": "billingAddress",
        "type": "ShopperInput.BillingAddressEnum"
    },
    {
        "name": "deliveryAddress",
        "baseName": "deliveryAddress",
        "type": "ShopperInput.DeliveryAddressEnum"
    },
    {
        "name": "personalDetails",
        "baseName": "personalDetails",
        "type": "ShopperInput.PersonalDetailsEnum"
    }
];
(function (ShopperInput) {
    let BillingAddressEnum;
    (function (BillingAddressEnum) {
        BillingAddressEnum[BillingAddressEnum["Editable"] = 'editable'] = "Editable";
        BillingAddressEnum[BillingAddressEnum["Hidden"] = 'hidden'] = "Hidden";
        BillingAddressEnum[BillingAddressEnum["ReadOnly"] = 'readOnly'] = "ReadOnly";
    })(BillingAddressEnum = ShopperInput.BillingAddressEnum || (ShopperInput.BillingAddressEnum = {}));
    let DeliveryAddressEnum;
    (function (DeliveryAddressEnum) {
        DeliveryAddressEnum[DeliveryAddressEnum["Editable"] = 'editable'] = "Editable";
        DeliveryAddressEnum[DeliveryAddressEnum["Hidden"] = 'hidden'] = "Hidden";
        DeliveryAddressEnum[DeliveryAddressEnum["ReadOnly"] = 'readOnly'] = "ReadOnly";
    })(DeliveryAddressEnum = ShopperInput.DeliveryAddressEnum || (ShopperInput.DeliveryAddressEnum = {}));
    let PersonalDetailsEnum;
    (function (PersonalDetailsEnum) {
        PersonalDetailsEnum[PersonalDetailsEnum["Editable"] = 'editable'] = "Editable";
        PersonalDetailsEnum[PersonalDetailsEnum["Hidden"] = 'hidden'] = "Hidden";
        PersonalDetailsEnum[PersonalDetailsEnum["ReadOnly"] = 'readOnly'] = "ReadOnly";
    })(PersonalDetailsEnum = ShopperInput.PersonalDetailsEnum || (ShopperInput.PersonalDetailsEnum = {}));
})(ShopperInput = exports.ShopperInput || (exports.ShopperInput = {}));
