// @adyen/api-library/lib/src/typings/checkout/split.js
"use strict";
/*
 * The version of the OpenAPI document: v69
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Split = void 0;
class Split {
    static getAttributeTypeMap() {
        return Split.attributeTypeMap;
    }
}
exports.Split = Split;
Split.discriminator = undefined;
Split.attributeTypeMap = [
    {
        "name": "account",
        "baseName": "account",
        "type": "string"
    },
    {
        "name": "amount",
        "baseName": "amount",
        "type": "SplitAmount"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "Split.TypeEnum"
    }
];
(function (Split) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["BalanceAccount"] = 'BalanceAccount'] = "BalanceAccount";
        TypeEnum[TypeEnum["Commission"] = 'Commission'] = "Commission";
        TypeEnum[TypeEnum["Default"] = 'Default'] = "Default";
        TypeEnum[TypeEnum["MarketPlace"] = 'MarketPlace'] = "MarketPlace";
        TypeEnum[TypeEnum["PaymentFee"] = 'PaymentFee'] = "PaymentFee";
        TypeEnum[TypeEnum["Remainder"] = 'Remainder'] = "Remainder";
        TypeEnum[TypeEnum["Vat"] = 'VAT'] = "Vat";
        TypeEnum[TypeEnum["Verification"] = 'Verification'] = "Verification";
    })(TypeEnum = Split.TypeEnum || (Split.TypeEnum = {}));
})(Split = exports.Split || (exports.Split = {}));
