// @adyen/api-library/lib/src/typings/checkout/storedPaymentMethodDetails.js
"use strict";
/*
 * The version of the OpenAPI document: v69
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StoredPaymentMethodDetails = void 0;
class StoredPaymentMethodDetails {
    static getAttributeTypeMap() {
        return StoredPaymentMethodDetails.attributeTypeMap;
    }
}
exports.StoredPaymentMethodDetails = StoredPaymentMethodDetails;
StoredPaymentMethodDetails.discriminator = undefined;
StoredPaymentMethodDetails.attributeTypeMap = [
    {
        "name": "recurringDetailReference",
        "baseName": "recurringDetailReference",
        "type": "string"
    },
    {
        "name": "storedPaymentMethodId",
        "baseName": "storedPaymentMethodId",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "StoredPaymentMethodDetails.TypeEnum"
    }
];
(function (StoredPaymentMethodDetails) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["BcmcMobile"] = 'bcmc_mobile'] = "BcmcMobile";
        TypeEnum[TypeEnum["BcmcMobileQr"] = 'bcmc_mobile_QR'] = "BcmcMobileQr";
        TypeEnum[TypeEnum["BcmcMobileApp"] = 'bcmc_mobile_app'] = "BcmcMobileApp";
        TypeEnum[TypeEnum["MomoWallet"] = 'momo_wallet'] = "MomoWallet";
        TypeEnum[TypeEnum["MomoWalletApp"] = 'momo_wallet_app'] = "MomoWalletApp";
        TypeEnum[TypeEnum["PaymayaWallet"] = 'paymaya_wallet'] = "PaymayaWallet";
        TypeEnum[TypeEnum["GrabpaySg"] = 'grabpay_SG'] = "GrabpaySg";
        TypeEnum[TypeEnum["GrabpayMy"] = 'grabpay_MY'] = "GrabpayMy";
        TypeEnum[TypeEnum["GrabpayTh"] = 'grabpay_TH'] = "GrabpayTh";
        TypeEnum[TypeEnum["GrabpayId"] = 'grabpay_ID'] = "GrabpayId";
        TypeEnum[TypeEnum["GrabpayVn"] = 'grabpay_VN'] = "GrabpayVn";
        TypeEnum[TypeEnum["GrabpayPh"] = 'grabpay_PH'] = "GrabpayPh";
        TypeEnum[TypeEnum["Oxxo"] = 'oxxo'] = "Oxxo";
        TypeEnum[TypeEnum["Gcash"] = 'gcash'] = "Gcash";
        TypeEnum[TypeEnum["Kakaopay"] = 'kakaopay'] = "Kakaopay";
        TypeEnum[TypeEnum["Truemoney"] = 'truemoney'] = "Truemoney";
    })(TypeEnum = StoredPaymentMethodDetails.TypeEnum || (StoredPaymentMethodDetails.TypeEnum = {}));
})(StoredPaymentMethodDetails = exports.StoredPaymentMethodDetails || (exports.StoredPaymentMethodDetails = {}));
