// @adyen/api-library/lib/src/typings/balancePlatform/accountHolderCapability.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountHolderCapability = void 0;
class AccountHolderCapability {
    static getAttributeTypeMap() {
        return AccountHolderCapability.attributeTypeMap;
    }
}
exports.AccountHolderCapability = AccountHolderCapability;
AccountHolderCapability.discriminator = undefined;
AccountHolderCapability.attributeTypeMap = [
    {
        "name": "allowed",
        "baseName": "allowed",
        "type": "boolean"
    },
    {
        "name": "allowedLevel",
        "baseName": "allowedLevel",
        "type": "AccountHolderCapability.AllowedLevelEnum"
    },
    {
        "name": "allowedSettings",
        "baseName": "allowedSettings",
        "type": "JSONObject"
    },
    {
        "name": "enabled",
        "baseName": "enabled",
        "type": "boolean"
    },
    {
        "name": "problems",
        "baseName": "problems",
        "type": "Array<CapabilityProblem>"
    },
    {
        "name": "requested",
        "baseName": "requested",
        "type": "boolean"
    },
    {
        "name": "requestedLevel",
        "baseName": "requestedLevel",
        "type": "AccountHolderCapability.RequestedLevelEnum"
    },
    {
        "name": "requestedSettings",
        "baseName": "requestedSettings",
        "type": "JSONObject"
    },
    {
        "name": "verificationStatus",
        "baseName": "verificationStatus",
        "type": "AccountHolderCapability.VerificationStatusEnum"
    }
];
(function (AccountHolderCapability) {
    let AllowedLevelEnum;
    (function (AllowedLevelEnum) {
        AllowedLevelEnum[AllowedLevelEnum["High"] = 'high'] = "High";
        AllowedLevelEnum[AllowedLevelEnum["Low"] = 'low'] = "Low";
        AllowedLevelEnum[AllowedLevelEnum["Medium"] = 'medium'] = "Medium";
        AllowedLevelEnum[AllowedLevelEnum["NotApplicable"] = 'notApplicable'] = "NotApplicable";
    })(AllowedLevelEnum = AccountHolderCapability.AllowedLevelEnum || (AccountHolderCapability.AllowedLevelEnum = {}));
    let RequestedLevelEnum;
    (function (RequestedLevelEnum) {
        RequestedLevelEnum[RequestedLevelEnum["High"] = 'high'] = "High";
        RequestedLevelEnum[RequestedLevelEnum["Low"] = 'low'] = "Low";
        RequestedLevelEnum[RequestedLevelEnum["Medium"] = 'medium'] = "Medium";
        RequestedLevelEnum[RequestedLevelEnum["NotApplicable"] = 'notApplicable'] = "NotApplicable";
    })(RequestedLevelEnum = AccountHolderCapability.RequestedLevelEnum || (AccountHolderCapability.RequestedLevelEnum = {}));
    let VerificationStatusEnum;
    (function (VerificationStatusEnum) {
        VerificationStatusEnum[VerificationStatusEnum["Invalid"] = 'invalid'] = "Invalid";
        VerificationStatusEnum[VerificationStatusEnum["Pending"] = 'pending'] = "Pending";
        VerificationStatusEnum[VerificationStatusEnum["Rejected"] = 'rejected'] = "Rejected";
        VerificationStatusEnum[VerificationStatusEnum["Valid"] = 'valid'] = "Valid";
    })(VerificationStatusEnum = AccountHolderCapability.VerificationStatusEnum || (AccountHolderCapability.VerificationStatusEnum = {}));
})(AccountHolderCapability = exports.AccountHolderCapability || (exports.AccountHolderCapability = {}));
