// @adyen/api-library/lib/src/typings/balancePlatform/duration.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Duration = void 0;
class Duration {
    static getAttributeTypeMap() {
        return Duration.attributeTypeMap;
    }
}
exports.Duration = Duration;
Duration.discriminator = undefined;
Duration.attributeTypeMap = [
    {
        "name": "unit",
        "baseName": "unit",
        "type": "Duration.UnitEnum"
    },
    {
        "name": "value",
        "baseName": "value",
        "type": "number"
    }
];
(function (Duration) {
    let UnitEnum;
    (function (UnitEnum) {
        UnitEnum[UnitEnum["Days"] = 'days'] = "Days";
        UnitEnum[UnitEnum["Hours"] = 'hours'] = "Hours";
        UnitEnum[UnitEnum["Minutes"] = 'minutes'] = "Minutes";
        UnitEnum[UnitEnum["Months"] = 'months'] = "Months";
        UnitEnum[UnitEnum["Weeks"] = 'weeks'] = "Weeks";
    })(UnitEnum = Duration.UnitEnum || (Duration.UnitEnum = {}));
})(Duration = exports.Duration || (exports.Duration = {}));
