// @adyen/api-library/lib/src/typings/balancePlatform/paymentInstrumentUpdateRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentInstrumentUpdateRequest = void 0;
class PaymentInstrumentUpdateRequest {
    static getAttributeTypeMap() {
        return PaymentInstrumentUpdateRequest.attributeTypeMap;
    }
}
exports.PaymentInstrumentUpdateRequest = PaymentInstrumentUpdateRequest;
PaymentInstrumentUpdateRequest.discriminator = undefined;
PaymentInstrumentUpdateRequest.attributeTypeMap = [
    {
        "name": "balanceAccountId",
        "baseName": "balanceAccountId",
        "type": "string"
    },
    {
        "name": "card",
        "baseName": "card",
        "type": "CardInfo"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "PaymentInstrumentUpdateRequest.StatusEnum"
    },
    {
        "name": "statusComment",
        "baseName": "statusComment",
        "type": "string"
    },
    {
        "name": "statusReason",
        "baseName": "statusReason",
        "type": "PaymentInstrumentUpdateRequest.StatusReasonEnum"
    }
];
(function (PaymentInstrumentUpdateRequest) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["Active"] = 'active'] = "Active";
        StatusEnum[StatusEnum["Closed"] = 'closed'] = "Closed";
        StatusEnum[StatusEnum["Inactive"] = 'inactive'] = "Inactive";
        StatusEnum[StatusEnum["Suspended"] = 'suspended'] = "Suspended";
    })(StatusEnum = PaymentInstrumentUpdateRequest.StatusEnum || (PaymentInstrumentUpdateRequest.StatusEnum = {}));
    let StatusReasonEnum;
    (function (StatusReasonEnum) {
        StatusReasonEnum[StatusReasonEnum["AccountClosure"] = 'accountClosure'] = "AccountClosure";
        StatusReasonEnum[StatusReasonEnum["Damaged"] = 'damaged'] = "Damaged";
        StatusReasonEnum[StatusReasonEnum["EndOfLife"] = 'endOfLife'] = "EndOfLife";
        StatusReasonEnum[StatusReasonEnum["Expired"] = 'expired'] = "Expired";
        StatusReasonEnum[StatusReasonEnum["Lost"] = 'lost'] = "Lost";
        StatusReasonEnum[StatusReasonEnum["Other"] = 'other'] = "Other";
        StatusReasonEnum[StatusReasonEnum["Stolen"] = 'stolen'] = "Stolen";
        StatusReasonEnum[StatusReasonEnum["SuspectedFraud"] = 'suspectedFraud'] = "SuspectedFraud";
    })(StatusReasonEnum = PaymentInstrumentUpdateRequest.StatusReasonEnum || (PaymentInstrumentUpdateRequest.StatusReasonEnum = {}));
})(PaymentInstrumentUpdateRequest = exports.PaymentInstrumentUpdateRequest || (exports.PaymentInstrumentUpdateRequest = {}));
