// @adyen/api-library/lib/src/typings/balancePlatform/paymentInstrument.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentInstrument = void 0;
class PaymentInstrument {
    static getAttributeTypeMap() {
        return PaymentInstrument.attributeTypeMap;
    }
}
exports.PaymentInstrument = PaymentInstrument;
PaymentInstrument.discriminator = undefined;
PaymentInstrument.attributeTypeMap = [
    {
        "name": "balanceAccountId",
        "baseName": "balanceAccountId",
        "type": "string"
    },
    {
        "name": "card",
        "baseName": "card",
        "type": "Card"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "issuingCountryCode",
        "baseName": "issuingCountryCode",
        "type": "string"
    },
    {
        "name": "paymentInstrumentGroupId",
        "baseName": "paymentInstrumentGroupId",
        "type": "string"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "PaymentInstrument.StatusEnum"
    },
    {
        "name": "statusReason",
        "baseName": "statusReason",
        "type": "PaymentInstrument.StatusReasonEnum"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "PaymentInstrument.TypeEnum"
    }
];
(function (PaymentInstrument) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["Active"] = 'active'] = "Active";
        StatusEnum[StatusEnum["Closed"] = 'closed'] = "Closed";
        StatusEnum[StatusEnum["Inactive"] = 'inactive'] = "Inactive";
        StatusEnum[StatusEnum["Suspended"] = 'suspended'] = "Suspended";
    })(StatusEnum = PaymentInstrument.StatusEnum || (PaymentInstrument.StatusEnum = {}));
    let StatusReasonEnum;
    (function (StatusReasonEnum) {
        StatusReasonEnum[StatusReasonEnum["AccountClosure"] = 'accountClosure'] = "AccountClosure";
        StatusReasonEnum[StatusReasonEnum["Damaged"] = 'damaged'] = "Damaged";
        StatusReasonEnum[StatusReasonEnum["EndOfLife"] = 'endOfLife'] = "EndOfLife";
        StatusReasonEnum[StatusReasonEnum["Expired"] = 'expired'] = "Expired";
        StatusReasonEnum[StatusReasonEnum["Lost"] = 'lost'] = "Lost";
        StatusReasonEnum[StatusReasonEnum["Other"] = 'other'] = "Other";
        StatusReasonEnum[StatusReasonEnum["Stolen"] = 'stolen'] = "Stolen";
        StatusReasonEnum[StatusReasonEnum["SuspectedFraud"] = 'suspectedFraud'] = "SuspectedFraud";
    })(StatusReasonEnum = PaymentInstrument.StatusReasonEnum || (PaymentInstrument.StatusReasonEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["Card"] = 'card'] = "Card";
    })(TypeEnum = PaymentInstrument.TypeEnum || (PaymentInstrument.TypeEnum = {}));
})(PaymentInstrument = exports.PaymentInstrument || (exports.PaymentInstrument = {}));
