// @adyen/api-library/lib/src/typings/balancePlatform/paymentInstrumentInfo.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentInstrumentInfo = void 0;
class PaymentInstrumentInfo {
    static getAttributeTypeMap() {
        return PaymentInstrumentInfo.attributeTypeMap;
    }
}
exports.PaymentInstrumentInfo = PaymentInstrumentInfo;
PaymentInstrumentInfo.discriminator = undefined;
PaymentInstrumentInfo.attributeTypeMap = [
    {
        "name": "balanceAccountId",
        "baseName": "balanceAccountId",
        "type": "string"
    },
    {
        "name": "card",
        "baseName": "card",
        "type": "CardInfo"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "issuingCountryCode",
        "baseName": "issuingCountryCode",
        "type": "string"
    },
    {
        "name": "paymentInstrumentGroupId",
        "baseName": "paymentInstrumentGroupId",
        "type": "string"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "PaymentInstrumentInfo.StatusEnum"
    },
    {
        "name": "statusReason",
        "baseName": "statusReason",
        "type": "PaymentInstrumentInfo.StatusReasonEnum"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "PaymentInstrumentInfo.TypeEnum"
    }
];
(function (PaymentInstrumentInfo) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["Active"] = 'active'] = "Active";
        StatusEnum[StatusEnum["Closed"] = 'closed'] = "Closed";
        StatusEnum[StatusEnum["Inactive"] = 'inactive'] = "Inactive";
        StatusEnum[StatusEnum["Suspended"] = 'suspended'] = "Suspended";
    })(StatusEnum = PaymentInstrumentInfo.StatusEnum || (PaymentInstrumentInfo.StatusEnum = {}));
    let StatusReasonEnum;
    (function (StatusReasonEnum) {
        StatusReasonEnum[StatusReasonEnum["AccountClosure"] = 'accountClosure'] = "AccountClosure";
        StatusReasonEnum[StatusReasonEnum["Damaged"] = 'damaged'] = "Damaged";
        StatusReasonEnum[StatusReasonEnum["EndOfLife"] = 'endOfLife'] = "EndOfLife";
        StatusReasonEnum[StatusReasonEnum["Expired"] = 'expired'] = "Expired";
        StatusReasonEnum[StatusReasonEnum["Lost"] = 'lost'] = "Lost";
        StatusReasonEnum[StatusReasonEnum["Other"] = 'other'] = "Other";
        StatusReasonEnum[StatusReasonEnum["Stolen"] = 'stolen'] = "Stolen";
        StatusReasonEnum[StatusReasonEnum["SuspectedFraud"] = 'suspectedFraud'] = "SuspectedFraud";
    })(StatusReasonEnum = PaymentInstrumentInfo.StatusReasonEnum || (PaymentInstrumentInfo.StatusReasonEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["Card"] = 'card'] = "Card";
    })(TypeEnum = PaymentInstrumentInfo.TypeEnum || (PaymentInstrumentInfo.TypeEnum = {}));
})(PaymentInstrumentInfo = exports.PaymentInstrumentInfo || (exports.PaymentInstrumentInfo = {}));
