// @adyen/api-library/lib/src/typings/balancePlatform/sweepConfigurationV2.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SweepConfigurationV2 = void 0;
class SweepConfigurationV2 {
    static getAttributeTypeMap() {
        return SweepConfigurationV2.attributeTypeMap;
    }
}
exports.SweepConfigurationV2 = SweepConfigurationV2;
SweepConfigurationV2.discriminator = undefined;
SweepConfigurationV2.attributeTypeMap = [
    {
        "name": "counterparty",
        "baseName": "counterparty",
        "type": "SweepCounterparty"
    },
    {
        "name": "currency",
        "baseName": "currency",
        "type": "string"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "schedule",
        "baseName": "schedule",
        "type": "SweepSchedule"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "SweepConfigurationV2.StatusEnum"
    },
    {
        "name": "sweepAmount",
        "baseName": "sweepAmount",
        "type": "Amount"
    },
    {
        "name": "targetAmount",
        "baseName": "targetAmount",
        "type": "Amount"
    },
    {
        "name": "triggerAmount",
        "baseName": "triggerAmount",
        "type": "Amount"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "SweepConfigurationV2.TypeEnum"
    }
];
(function (SweepConfigurationV2) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["Active"] = 'active'] = "Active";
        StatusEnum[StatusEnum["Inactive"] = 'inactive'] = "Inactive";
    })(StatusEnum = SweepConfigurationV2.StatusEnum || (SweepConfigurationV2.StatusEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["Pull"] = 'pull'] = "Pull";
        TypeEnum[TypeEnum["Push"] = 'push'] = "Push";
    })(TypeEnum = SweepConfigurationV2.TypeEnum || (SweepConfigurationV2.TypeEnum = {}));
})(SweepConfigurationV2 = exports.SweepConfigurationV2 || (exports.SweepConfigurationV2 = {}));
