// @adyen/api-library/lib/src/typings/balancePlatform/transactionRuleInterval.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransactionRuleInterval = void 0;
class TransactionRuleInterval {
    static getAttributeTypeMap() {
        return TransactionRuleInterval.attributeTypeMap;
    }
}
exports.TransactionRuleInterval = TransactionRuleInterval;
TransactionRuleInterval.discriminator = undefined;
TransactionRuleInterval.attributeTypeMap = [
    {
        "name": "dayOfMonth",
        "baseName": "dayOfMonth",
        "type": "number"
    },
    {
        "name": "dayOfWeek",
        "baseName": "dayOfWeek",
        "type": "TransactionRuleInterval.DayOfWeekEnum"
    },
    {
        "name": "duration",
        "baseName": "duration",
        "type": "Duration"
    },
    {
        "name": "timeOfDay",
        "baseName": "timeOfDay",
        "type": "string"
    },
    {
        "name": "timeZone",
        "baseName": "timeZone",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "TransactionRuleInterval.TypeEnum"
    }
];
(function (TransactionRuleInterval) {
    let DayOfWeekEnum;
    (function (DayOfWeekEnum) {
        DayOfWeekEnum[DayOfWeekEnum["Friday"] = 'friday'] = "Friday";
        DayOfWeekEnum[DayOfWeekEnum["Monday"] = 'monday'] = "Monday";
        DayOfWeekEnum[DayOfWeekEnum["Saturday"] = 'saturday'] = "Saturday";
        DayOfWeekEnum[DayOfWeekEnum["Sunday"] = 'sunday'] = "Sunday";
        DayOfWeekEnum[DayOfWeekEnum["Thursday"] = 'thursday'] = "Thursday";
        DayOfWeekEnum[DayOfWeekEnum["Tuesday"] = 'tuesday'] = "Tuesday";
        DayOfWeekEnum[DayOfWeekEnum["Wednesday"] = 'wednesday'] = "Wednesday";
    })(DayOfWeekEnum = TransactionRuleInterval.DayOfWeekEnum || (TransactionRuleInterval.DayOfWeekEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["Daily"] = 'daily'] = "Daily";
        TypeEnum[TypeEnum["Lifetime"] = 'lifetime'] = "Lifetime";
        TypeEnum[TypeEnum["Monthly"] = 'monthly'] = "Monthly";
        TypeEnum[TypeEnum["PerTransaction"] = 'perTransaction'] = "PerTransaction";
        TypeEnum[TypeEnum["Rolling"] = 'rolling'] = "Rolling";
        TypeEnum[TypeEnum["Sliding"] = 'sliding'] = "Sliding";
        TypeEnum[TypeEnum["Weekly"] = 'weekly'] = "Weekly";
    })(TypeEnum = TransactionRuleInterval.TypeEnum || (TransactionRuleInterval.TypeEnum = {}));
})(TransactionRuleInterval = exports.TransactionRuleInterval || (exports.TransactionRuleInterval = {}));
