// @adyen/api-library/lib/src/typings/balancePlatform/transactionRule.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransactionRule = void 0;
class TransactionRule {
    static getAttributeTypeMap() {
        return TransactionRule.attributeTypeMap;
    }
}
exports.TransactionRule = TransactionRule;
TransactionRule.discriminator = undefined;
TransactionRule.attributeTypeMap = [
    {
        "name": "aggregationLevel",
        "baseName": "aggregationLevel",
        "type": "string"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "endDate",
        "baseName": "endDate",
        "type": "string"
    },
    {
        "name": "entityKey",
        "baseName": "entityKey",
        "type": "TransactionRuleEntityKey"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "interval",
        "baseName": "interval",
        "type": "TransactionRuleInterval"
    },
    {
        "name": "outcomeType",
        "baseName": "outcomeType",
        "type": "TransactionRule.OutcomeTypeEnum"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "ruleRestrictions",
        "baseName": "ruleRestrictions",
        "type": "TransactionRuleRestrictions"
    },
    {
        "name": "score",
        "baseName": "score",
        "type": "number"
    },
    {
        "name": "startDate",
        "baseName": "startDate",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "TransactionRule.StatusEnum"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "TransactionRule.TypeEnum"
    }
];
(function (TransactionRule) {
    let OutcomeTypeEnum;
    (function (OutcomeTypeEnum) {
        OutcomeTypeEnum[OutcomeTypeEnum["HardBlock"] = 'hardBlock'] = "HardBlock";
        OutcomeTypeEnum[OutcomeTypeEnum["ScoreBased"] = 'scoreBased'] = "ScoreBased";
    })(OutcomeTypeEnum = TransactionRule.OutcomeTypeEnum || (TransactionRule.OutcomeTypeEnum = {}));
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["Active"] = 'active'] = "Active";
        StatusEnum[StatusEnum["Inactive"] = 'inactive'] = "Inactive";
    })(StatusEnum = TransactionRule.StatusEnum || (TransactionRule.StatusEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["AllowList"] = 'allowList'] = "AllowList";
        TypeEnum[TypeEnum["BlockList"] = 'blockList'] = "BlockList";
        TypeEnum[TypeEnum["MaxUsage"] = 'maxUsage'] = "MaxUsage";
        TypeEnum[TypeEnum["Velocity"] = 'velocity'] = "Velocity";
    })(TypeEnum = TransactionRule.TypeEnum || (TransactionRule.TypeEnum = {}));
})(TransactionRule = exports.TransactionRule || (exports.TransactionRule = {}));
