// @adyen/api-library/lib/src/typings/balancePlatform/verificationErrorRecursive.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.VerificationErrorRecursive = void 0;
class VerificationErrorRecursive {
    static getAttributeTypeMap() {
        return VerificationErrorRecursive.attributeTypeMap;
    }
}
exports.VerificationErrorRecursive = VerificationErrorRecursive;
VerificationErrorRecursive.discriminator = undefined;
VerificationErrorRecursive.attributeTypeMap = [
    {
        "name": "code",
        "baseName": "code",
        "type": "string"
    },
    {
        "name": "message",
        "baseName": "message",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "VerificationErrorRecursive.TypeEnum"
    },
    {
        "name": "remediatingActions",
        "baseName": "remediatingActions",
        "type": "Array<RemediatingAction>"
    }
];
(function (VerificationErrorRecursive) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["DataMissing"] = 'dataMissing'] = "DataMissing";
        TypeEnum[TypeEnum["InvalidInput"] = 'invalidInput'] = "InvalidInput";
        TypeEnum[TypeEnum["PendingStatus"] = 'pendingStatus'] = "PendingStatus";
    })(TypeEnum = VerificationErrorRecursive.TypeEnum || (VerificationErrorRecursive.TypeEnum = {}));
})(VerificationErrorRecursive = exports.VerificationErrorRecursive || (exports.VerificationErrorRecursive = {}));
