// @adyen/api-library/lib/src/typings/platformsFund/accountHolderTransactionListRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountHolderTransactionListRequest = void 0;
class AccountHolderTransactionListRequest {
    static getAttributeTypeMap() {
        return AccountHolderTransactionListRequest.attributeTypeMap;
    }
}
exports.AccountHolderTransactionListRequest = AccountHolderTransactionListRequest;
AccountHolderTransactionListRequest.discriminator = undefined;
AccountHolderTransactionListRequest.attributeTypeMap = [
    {
        "name": "accountHolderCode",
        "baseName": "accountHolderCode",
        "type": "string"
    },
    {
        "name": "transactionListsPerAccount",
        "baseName": "transactionListsPerAccount",
        "type": "Array<TransactionListForAccount>"
    },
    {
        "name": "transactionStatuses",
        "baseName": "transactionStatuses",
        "type": "Array<AccountHolderTransactionListRequest.TransactionStatusesEnum>"
    }
];
(function (AccountHolderTransactionListRequest) {
    let TransactionStatusesEnum;
    (function (TransactionStatusesEnum) {
        TransactionStatusesEnum[TransactionStatusesEnum["BalanceNotPaidOutTransfer"] = 'BalanceNotPaidOutTransfer'] = "BalanceNotPaidOutTransfer";
        TransactionStatusesEnum[TransactionStatusesEnum["Chargeback"] = 'Chargeback'] = "Chargeback";
        TransactionStatusesEnum[TransactionStatusesEnum["ChargebackCorrection"] = 'ChargebackCorrection'] = "ChargebackCorrection";
        TransactionStatusesEnum[TransactionStatusesEnum["ChargebackCorrectionReceived"] = 'ChargebackCorrectionReceived'] = "ChargebackCorrectionReceived";
        TransactionStatusesEnum[TransactionStatusesEnum["ChargebackReceived"] = 'ChargebackReceived'] = "ChargebackReceived";
        TransactionStatusesEnum[TransactionStatusesEnum["ChargebackReversed"] = 'ChargebackReversed'] = "ChargebackReversed";
        TransactionStatusesEnum[TransactionStatusesEnum["ChargebackReversedCorrection"] = 'ChargebackReversedCorrection'] = "ChargebackReversedCorrection";
        TransactionStatusesEnum[TransactionStatusesEnum["ChargebackReversedCorrectionReceived"] = 'ChargebackReversedCorrectionReceived'] = "ChargebackReversedCorrectionReceived";
        TransactionStatusesEnum[TransactionStatusesEnum["ChargebackReversedReceived"] = 'ChargebackReversedReceived'] = "ChargebackReversedReceived";
        TransactionStatusesEnum[TransactionStatusesEnum["Converted"] = 'Converted'] = "Converted";
        TransactionStatusesEnum[TransactionStatusesEnum["CreditClosed"] = 'CreditClosed'] = "CreditClosed";
        TransactionStatusesEnum[TransactionStatusesEnum["CreditFailed"] = 'CreditFailed'] = "CreditFailed";
        TransactionStatusesEnum[TransactionStatusesEnum["CreditReversed"] = 'CreditReversed'] = "CreditReversed";
        TransactionStatusesEnum[TransactionStatusesEnum["CreditReversedReceived"] = 'CreditReversedReceived'] = "CreditReversedReceived";
        TransactionStatusesEnum[TransactionStatusesEnum["CreditSuspended"] = 'CreditSuspended'] = "CreditSuspended";
        TransactionStatusesEnum[TransactionStatusesEnum["Credited"] = 'Credited'] = "Credited";
        TransactionStatusesEnum[TransactionStatusesEnum["DebitFailed"] = 'DebitFailed'] = "DebitFailed";
        TransactionStatusesEnum[TransactionStatusesEnum["DebitReversedReceived"] = 'DebitReversedReceived'] = "DebitReversedReceived";
        TransactionStatusesEnum[TransactionStatusesEnum["Debited"] = 'Debited'] = "Debited";
        TransactionStatusesEnum[TransactionStatusesEnum["DebitedReversed"] = 'DebitedReversed'] = "DebitedReversed";
        TransactionStatusesEnum[TransactionStatusesEnum["DepositCorrectionCredited"] = 'DepositCorrectionCredited'] = "DepositCorrectionCredited";
        TransactionStatusesEnum[TransactionStatusesEnum["DepositCorrectionDebited"] = 'DepositCorrectionDebited'] = "DepositCorrectionDebited";
        TransactionStatusesEnum[TransactionStatusesEnum["Fee"] = 'Fee'] = "Fee";
        TransactionStatusesEnum[TransactionStatusesEnum["FundTransfer"] = 'FundTransfer'] = "FundTransfer";
        TransactionStatusesEnum[TransactionStatusesEnum["FundTransferReversed"] = 'FundTransferReversed'] = "FundTransferReversed";
        TransactionStatusesEnum[TransactionStatusesEnum["InvoiceDeductionCredited"] = 'InvoiceDeductionCredited'] = "InvoiceDeductionCredited";
        TransactionStatusesEnum[TransactionStatusesEnum["InvoiceDeductionDebited"] = 'InvoiceDeductionDebited'] = "InvoiceDeductionDebited";
        TransactionStatusesEnum[TransactionStatusesEnum["ManualCorrected"] = 'ManualCorrected'] = "ManualCorrected";
        TransactionStatusesEnum[TransactionStatusesEnum["ManualCorrectionCredited"] = 'ManualCorrectionCredited'] = "ManualCorrectionCredited";
        TransactionStatusesEnum[TransactionStatusesEnum["ManualCorrectionDebited"] = 'ManualCorrectionDebited'] = "ManualCorrectionDebited";
        TransactionStatusesEnum[TransactionStatusesEnum["MerchantPayin"] = 'MerchantPayin'] = "MerchantPayin";
        TransactionStatusesEnum[TransactionStatusesEnum["MerchantPayinReversed"] = 'MerchantPayinReversed'] = "MerchantPayinReversed";
        TransactionStatusesEnum[TransactionStatusesEnum["Payout"] = 'Payout'] = "Payout";
        TransactionStatusesEnum[TransactionStatusesEnum["PayoutReversed"] = 'PayoutReversed'] = "PayoutReversed";
        TransactionStatusesEnum[TransactionStatusesEnum["PendingCredit"] = 'PendingCredit'] = "PendingCredit";
        TransactionStatusesEnum[TransactionStatusesEnum["PendingDebit"] = 'PendingDebit'] = "PendingDebit";
        TransactionStatusesEnum[TransactionStatusesEnum["PendingFundTransfer"] = 'PendingFundTransfer'] = "PendingFundTransfer";
        TransactionStatusesEnum[TransactionStatusesEnum["ReCredited"] = 'ReCredited'] = "ReCredited";
        TransactionStatusesEnum[TransactionStatusesEnum["ReCreditedReceived"] = 'ReCreditedReceived'] = "ReCreditedReceived";
        TransactionStatusesEnum[TransactionStatusesEnum["SecondChargeback"] = 'SecondChargeback'] = "SecondChargeback";
        TransactionStatusesEnum[TransactionStatusesEnum["SecondChargebackCorrection"] = 'SecondChargebackCorrection'] = "SecondChargebackCorrection";
        TransactionStatusesEnum[TransactionStatusesEnum["SecondChargebackCorrectionReceived"] = 'SecondChargebackCorrectionReceived'] = "SecondChargebackCorrectionReceived";
        TransactionStatusesEnum[TransactionStatusesEnum["SecondChargebackReceived"] = 'SecondChargebackReceived'] = "SecondChargebackReceived";
    })(TransactionStatusesEnum = AccountHolderTransactionListRequest.TransactionStatusesEnum || (AccountHolderTransactionListRequest.TransactionStatusesEnum = {}));
})(AccountHolderTransactionListRequest = exports.AccountHolderTransactionListRequest || (exports.AccountHolderTransactionListRequest = {}));
