// @adyen/api-library/lib/src/typings/platformsFund/transaction.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Transaction = void 0;
class Transaction {
    static getAttributeTypeMap() {
        return Transaction.attributeTypeMap;
    }
}
exports.Transaction = Transaction;
Transaction.discriminator = undefined;
Transaction.attributeTypeMap = [
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount"
    },
    {
        "name": "bankAccountDetail",
        "baseName": "bankAccountDetail",
        "type": "BankAccountDetail"
    },
    {
        "name": "captureMerchantReference",
        "baseName": "captureMerchantReference",
        "type": "string"
    },
    {
        "name": "capturePspReference",
        "baseName": "capturePspReference",
        "type": "string"
    },
    {
        "name": "creationDate",
        "baseName": "creationDate",
        "type": "Date"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "destinationAccountCode",
        "baseName": "destinationAccountCode",
        "type": "string"
    },
    {
        "name": "disputePspReference",
        "baseName": "disputePspReference",
        "type": "string"
    },
    {
        "name": "disputeReasonCode",
        "baseName": "disputeReasonCode",
        "type": "string"
    },
    {
        "name": "merchantReference",
        "baseName": "merchantReference",
        "type": "string"
    },
    {
        "name": "paymentPspReference",
        "baseName": "paymentPspReference",
        "type": "string"
    },
    {
        "name": "payoutPspReference",
        "baseName": "payoutPspReference",
        "type": "string"
    },
    {
        "name": "pspReference",
        "baseName": "pspReference",
        "type": "string"
    },
    {
        "name": "sourceAccountCode",
        "baseName": "sourceAccountCode",
        "type": "string"
    },
    {
        "name": "transactionStatus",
        "baseName": "transactionStatus",
        "type": "Transaction.TransactionStatusEnum"
    },
    {
        "name": "transferCode",
        "baseName": "transferCode",
        "type": "string"
    }
];
(function (Transaction) {
    let TransactionStatusEnum;
    (function (TransactionStatusEnum) {
        TransactionStatusEnum[TransactionStatusEnum["BalanceNotPaidOutTransfer"] = 'BalanceNotPaidOutTransfer'] = "BalanceNotPaidOutTransfer";
        TransactionStatusEnum[TransactionStatusEnum["Chargeback"] = 'Chargeback'] = "Chargeback";
        TransactionStatusEnum[TransactionStatusEnum["ChargebackCorrection"] = 'ChargebackCorrection'] = "ChargebackCorrection";
        TransactionStatusEnum[TransactionStatusEnum["ChargebackCorrectionReceived"] = 'ChargebackCorrectionReceived'] = "ChargebackCorrectionReceived";
        TransactionStatusEnum[TransactionStatusEnum["ChargebackReceived"] = 'ChargebackReceived'] = "ChargebackReceived";
        TransactionStatusEnum[TransactionStatusEnum["ChargebackReversed"] = 'ChargebackReversed'] = "ChargebackReversed";
        TransactionStatusEnum[TransactionStatusEnum["ChargebackReversedCorrection"] = 'ChargebackReversedCorrection'] = "ChargebackReversedCorrection";
        TransactionStatusEnum[TransactionStatusEnum["ChargebackReversedCorrectionReceived"] = 'ChargebackReversedCorrectionReceived'] = "ChargebackReversedCorrectionReceived";
        TransactionStatusEnum[TransactionStatusEnum["ChargebackReversedReceived"] = 'ChargebackReversedReceived'] = "ChargebackReversedReceived";
        TransactionStatusEnum[TransactionStatusEnum["Converted"] = 'Converted'] = "Converted";
        TransactionStatusEnum[TransactionStatusEnum["CreditClosed"] = 'CreditClosed'] = "CreditClosed";
        TransactionStatusEnum[TransactionStatusEnum["CreditFailed"] = 'CreditFailed'] = "CreditFailed";
        TransactionStatusEnum[TransactionStatusEnum["CreditReversed"] = 'CreditReversed'] = "CreditReversed";
        TransactionStatusEnum[TransactionStatusEnum["CreditReversedReceived"] = 'CreditReversedReceived'] = "CreditReversedReceived";
        TransactionStatusEnum[TransactionStatusEnum["CreditSuspended"] = 'CreditSuspended'] = "CreditSuspended";
        TransactionStatusEnum[TransactionStatusEnum["Credited"] = 'Credited'] = "Credited";
        TransactionStatusEnum[TransactionStatusEnum["DebitFailed"] = 'DebitFailed'] = "DebitFailed";
        TransactionStatusEnum[TransactionStatusEnum["DebitReversedReceived"] = 'DebitReversedReceived'] = "DebitReversedReceived";
        TransactionStatusEnum[TransactionStatusEnum["Debited"] = 'Debited'] = "Debited";
        TransactionStatusEnum[TransactionStatusEnum["DebitedReversed"] = 'DebitedReversed'] = "DebitedReversed";
        TransactionStatusEnum[TransactionStatusEnum["DepositCorrectionCredited"] = 'DepositCorrectionCredited'] = "DepositCorrectionCredited";
        TransactionStatusEnum[TransactionStatusEnum["DepositCorrectionDebited"] = 'DepositCorrectionDebited'] = "DepositCorrectionDebited";
        TransactionStatusEnum[TransactionStatusEnum["Fee"] = 'Fee'] = "Fee";
        TransactionStatusEnum[TransactionStatusEnum["FundTransfer"] = 'FundTransfer'] = "FundTransfer";
        TransactionStatusEnum[TransactionStatusEnum["FundTransferReversed"] = 'FundTransferReversed'] = "FundTransferReversed";
        TransactionStatusEnum[TransactionStatusEnum["InvoiceDeductionCredited"] = 'InvoiceDeductionCredited'] = "InvoiceDeductionCredited";
        TransactionStatusEnum[TransactionStatusEnum["InvoiceDeductionDebited"] = 'InvoiceDeductionDebited'] = "InvoiceDeductionDebited";
        TransactionStatusEnum[TransactionStatusEnum["ManualCorrected"] = 'ManualCorrected'] = "ManualCorrected";
        TransactionStatusEnum[TransactionStatusEnum["ManualCorrectionCredited"] = 'ManualCorrectionCredited'] = "ManualCorrectionCredited";
        TransactionStatusEnum[TransactionStatusEnum["ManualCorrectionDebited"] = 'ManualCorrectionDebited'] = "ManualCorrectionDebited";
        TransactionStatusEnum[TransactionStatusEnum["MerchantPayin"] = 'MerchantPayin'] = "MerchantPayin";
        TransactionStatusEnum[TransactionStatusEnum["MerchantPayinReversed"] = 'MerchantPayinReversed'] = "MerchantPayinReversed";
        TransactionStatusEnum[TransactionStatusEnum["Payout"] = 'Payout'] = "Payout";
        TransactionStatusEnum[TransactionStatusEnum["PayoutReversed"] = 'PayoutReversed'] = "PayoutReversed";
        TransactionStatusEnum[TransactionStatusEnum["PendingCredit"] = 'PendingCredit'] = "PendingCredit";
        TransactionStatusEnum[TransactionStatusEnum["PendingDebit"] = 'PendingDebit'] = "PendingDebit";
        TransactionStatusEnum[TransactionStatusEnum["PendingFundTransfer"] = 'PendingFundTransfer'] = "PendingFundTransfer";
        TransactionStatusEnum[TransactionStatusEnum["ReCredited"] = 'ReCredited'] = "ReCredited";
        TransactionStatusEnum[TransactionStatusEnum["ReCreditedReceived"] = 'ReCreditedReceived'] = "ReCreditedReceived";
        TransactionStatusEnum[TransactionStatusEnum["SecondChargeback"] = 'SecondChargeback'] = "SecondChargeback";
        TransactionStatusEnum[TransactionStatusEnum["SecondChargebackCorrection"] = 'SecondChargebackCorrection'] = "SecondChargebackCorrection";
        TransactionStatusEnum[TransactionStatusEnum["SecondChargebackCorrectionReceived"] = 'SecondChargebackCorrectionReceived'] = "SecondChargebackCorrectionReceived";
        TransactionStatusEnum[TransactionStatusEnum["SecondChargebackReceived"] = 'SecondChargebackReceived'] = "SecondChargebackReceived";
    })(TransactionStatusEnum = Transaction.TransactionStatusEnum || (Transaction.TransactionStatusEnum = {}));
})(Transaction = exports.Transaction || (exports.Transaction = {}));
