// @adyen/api-library/lib/src/typings/platformsAccount/createAccountRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateAccountRequest = void 0;
class CreateAccountRequest {
    static getAttributeTypeMap() {
        return CreateAccountRequest.attributeTypeMap;
    }
}
exports.CreateAccountRequest = CreateAccountRequest;
CreateAccountRequest.discriminator = undefined;
CreateAccountRequest.attributeTypeMap = [
    {
        "name": "accountHolderCode",
        "baseName": "accountHolderCode",
        "type": "string"
    },
    {
        "name": "bankAccountUUID",
        "baseName": "bankAccountUUID",
        "type": "string"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "metadata",
        "baseName": "metadata",
        "type": "{ [key: string]: string; }"
    },
    {
        "name": "payoutMethodCode",
        "baseName": "payoutMethodCode",
        "type": "string"
    },
    {
        "name": "payoutSchedule",
        "baseName": "payoutSchedule",
        "type": "CreateAccountRequest.PayoutScheduleEnum"
    },
    {
        "name": "payoutScheduleReason",
        "baseName": "payoutScheduleReason",
        "type": "string"
    },
    {
        "name": "payoutSpeed",
        "baseName": "payoutSpeed",
        "type": "CreateAccountRequest.PayoutSpeedEnum"
    }
];
(function (CreateAccountRequest) {
    let PayoutScheduleEnum;
    (function (PayoutScheduleEnum) {
        PayoutScheduleEnum[PayoutScheduleEnum["BiweeklyOn1StAnd15ThAtMidnight"] = 'BIWEEKLY_ON_1ST_AND_15TH_AT_MIDNIGHT'] = "BiweeklyOn1StAnd15ThAtMidnight";
        PayoutScheduleEnum[PayoutScheduleEnum["BiweeklyOn1StAnd15ThAtNoon"] = 'BIWEEKLY_ON_1ST_AND_15TH_AT_NOON'] = "BiweeklyOn1StAnd15ThAtNoon";
        PayoutScheduleEnum[PayoutScheduleEnum["BiDailyAu"] = 'BI_DAILY_AU'] = "BiDailyAu";
        PayoutScheduleEnum[PayoutScheduleEnum["BiDailyEu"] = 'BI_DAILY_EU'] = "BiDailyEu";
        PayoutScheduleEnum[PayoutScheduleEnum["BiDailyUs"] = 'BI_DAILY_US'] = "BiDailyUs";
        PayoutScheduleEnum[PayoutScheduleEnum["Daily"] = 'DAILY'] = "Daily";
        PayoutScheduleEnum[PayoutScheduleEnum["Daily6Pm"] = 'DAILY_6PM'] = "Daily6Pm";
        PayoutScheduleEnum[PayoutScheduleEnum["DailyAu"] = 'DAILY_AU'] = "DailyAu";
        PayoutScheduleEnum[PayoutScheduleEnum["DailyEu"] = 'DAILY_EU'] = "DailyEu";
        PayoutScheduleEnum[PayoutScheduleEnum["DailySg"] = 'DAILY_SG'] = "DailySg";
        PayoutScheduleEnum[PayoutScheduleEnum["DailyUs"] = 'DAILY_US'] = "DailyUs";
        PayoutScheduleEnum[PayoutScheduleEnum["Default"] = 'DEFAULT'] = "Default";
        PayoutScheduleEnum[PayoutScheduleEnum["Every6HoursFromMidnight"] = 'EVERY_6_HOURS_FROM_MIDNIGHT'] = "Every6HoursFromMidnight";
        PayoutScheduleEnum[PayoutScheduleEnum["Hold"] = 'HOLD'] = "Hold";
        PayoutScheduleEnum[PayoutScheduleEnum["Monthly"] = 'MONTHLY'] = "Monthly";
        PayoutScheduleEnum[PayoutScheduleEnum["MonthlyOn15ThAtMidnight"] = 'MONTHLY_ON_15TH_AT_MIDNIGHT'] = "MonthlyOn15ThAtMidnight";
        PayoutScheduleEnum[PayoutScheduleEnum["Weekly"] = 'WEEKLY'] = "Weekly";
        PayoutScheduleEnum[PayoutScheduleEnum["WeeklyMonToFriAu"] = 'WEEKLY_MON_TO_FRI_AU'] = "WeeklyMonToFriAu";
        PayoutScheduleEnum[PayoutScheduleEnum["WeeklyMonToFriEu"] = 'WEEKLY_MON_TO_FRI_EU'] = "WeeklyMonToFriEu";
        PayoutScheduleEnum[PayoutScheduleEnum["WeeklyMonToFriUs"] = 'WEEKLY_MON_TO_FRI_US'] = "WeeklyMonToFriUs";
        PayoutScheduleEnum[PayoutScheduleEnum["WeeklyOnTueFriMidnight"] = 'WEEKLY_ON_TUE_FRI_MIDNIGHT'] = "WeeklyOnTueFriMidnight";
        PayoutScheduleEnum[PayoutScheduleEnum["WeeklySunToThuAu"] = 'WEEKLY_SUN_TO_THU_AU'] = "WeeklySunToThuAu";
        PayoutScheduleEnum[PayoutScheduleEnum["WeeklySunToThuUs"] = 'WEEKLY_SUN_TO_THU_US'] = "WeeklySunToThuUs";
        PayoutScheduleEnum[PayoutScheduleEnum["Yearly"] = 'YEARLY'] = "Yearly";
    })(PayoutScheduleEnum = CreateAccountRequest.PayoutScheduleEnum || (CreateAccountRequest.PayoutScheduleEnum = {}));
    let PayoutSpeedEnum;
    (function (PayoutSpeedEnum) {
        PayoutSpeedEnum[PayoutSpeedEnum["Instant"] = 'INSTANT'] = "Instant";
        PayoutSpeedEnum[PayoutSpeedEnum["SameDay"] = 'SAME_DAY'] = "SameDay";
        PayoutSpeedEnum[PayoutSpeedEnum["Standard"] = 'STANDARD'] = "Standard";
    })(PayoutSpeedEnum = CreateAccountRequest.PayoutSpeedEnum || (CreateAccountRequest.PayoutSpeedEnum = {}));
})(CreateAccountRequest = exports.CreateAccountRequest || (exports.CreateAccountRequest = {}));
