// @adyen/api-library/lib/src/typings/platformsAccount/documentDetail.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DocumentDetail = void 0;
class DocumentDetail {
    static getAttributeTypeMap() {
        return DocumentDetail.attributeTypeMap;
    }
}
exports.DocumentDetail = DocumentDetail;
DocumentDetail.discriminator = undefined;
DocumentDetail.attributeTypeMap = [
    {
        "name": "accountHolderCode",
        "baseName": "accountHolderCode",
        "type": "string"
    },
    {
        "name": "bankAccountUUID",
        "baseName": "bankAccountUUID",
        "type": "string"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "documentType",
        "baseName": "documentType",
        "type": "DocumentDetail.DocumentTypeEnum"
    },
    {
        "name": "filename",
        "baseName": "filename",
        "type": "string"
    },
    {
        "name": "legalArrangementCode",
        "baseName": "legalArrangementCode",
        "type": "string"
    },
    {
        "name": "legalArrangementEntityCode",
        "baseName": "legalArrangementEntityCode",
        "type": "string"
    },
    {
        "name": "shareholderCode",
        "baseName": "shareholderCode",
        "type": "string"
    },
    {
        "name": "signatoryCode",
        "baseName": "signatoryCode",
        "type": "string"
    }
];
(function (DocumentDetail) {
    let DocumentTypeEnum;
    (function (DocumentTypeEnum) {
        DocumentTypeEnum[DocumentTypeEnum["BankStatement"] = 'BANK_STATEMENT'] = "BankStatement";
        DocumentTypeEnum[DocumentTypeEnum["Bsn"] = 'BSN'] = "Bsn";
        DocumentTypeEnum[DocumentTypeEnum["CompanyRegistrationScreening"] = 'COMPANY_REGISTRATION_SCREENING'] = "CompanyRegistrationScreening";
        DocumentTypeEnum[DocumentTypeEnum["ConstitutionalDocument"] = 'CONSTITUTIONAL_DOCUMENT'] = "ConstitutionalDocument";
        DocumentTypeEnum[DocumentTypeEnum["DrivingLicence"] = 'DRIVING_LICENCE'] = "DrivingLicence";
        DocumentTypeEnum[DocumentTypeEnum["DrivingLicenceBack"] = 'DRIVING_LICENCE_BACK'] = "DrivingLicenceBack";
        DocumentTypeEnum[DocumentTypeEnum["DrivingLicenceFront"] = 'DRIVING_LICENCE_FRONT'] = "DrivingLicenceFront";
        DocumentTypeEnum[DocumentTypeEnum["IdCard"] = 'ID_CARD'] = "IdCard";
        DocumentTypeEnum[DocumentTypeEnum["IdCardBack"] = 'ID_CARD_BACK'] = "IdCardBack";
        DocumentTypeEnum[DocumentTypeEnum["IdCardFront"] = 'ID_CARD_FRONT'] = "IdCardFront";
        DocumentTypeEnum[DocumentTypeEnum["Passport"] = 'PASSPORT'] = "Passport";
        DocumentTypeEnum[DocumentTypeEnum["ProofOfResidency"] = 'PROOF_OF_RESIDENCY'] = "ProofOfResidency";
        DocumentTypeEnum[DocumentTypeEnum["Ssn"] = 'SSN'] = "Ssn";
        DocumentTypeEnum[DocumentTypeEnum["SupportingDocuments"] = 'SUPPORTING_DOCUMENTS'] = "SupportingDocuments";
    })(DocumentTypeEnum = DocumentDetail.DocumentTypeEnum || (DocumentDetail.DocumentTypeEnum = {}));
})(DocumentDetail = exports.DocumentDetail || (exports.DocumentDetail = {}));
