// @adyen/api-library/lib/src/typings/platformsAccount/kYCCheckStatusData.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.KYCCheckStatusData = void 0;
class KYCCheckStatusData {
    static getAttributeTypeMap() {
        return KYCCheckStatusData.attributeTypeMap;
    }
}
exports.KYCCheckStatusData = KYCCheckStatusData;
KYCCheckStatusData.discriminator = undefined;
KYCCheckStatusData.attributeTypeMap = [
    {
        "name": "requiredFields",
        "baseName": "requiredFields",
        "type": "Array<string>"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "KYCCheckStatusData.StatusEnum"
    },
    {
        "name": "summary",
        "baseName": "summary",
        "type": "KYCCheckSummary"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "KYCCheckStatusData.TypeEnum"
    }
];
(function (KYCCheckStatusData) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["AwaitingData"] = 'AWAITING_DATA'] = "AwaitingData";
        StatusEnum[StatusEnum["DataProvided"] = 'DATA_PROVIDED'] = "DataProvided";
        StatusEnum[StatusEnum["Failed"] = 'FAILED'] = "Failed";
        StatusEnum[StatusEnum["InvalidData"] = 'INVALID_DATA'] = "InvalidData";
        StatusEnum[StatusEnum["Passed"] = 'PASSED'] = "Passed";
        StatusEnum[StatusEnum["Pending"] = 'PENDING'] = "Pending";
        StatusEnum[StatusEnum["PendingReview"] = 'PENDING_REVIEW'] = "PendingReview";
        StatusEnum[StatusEnum["RetryLimitReached"] = 'RETRY_LIMIT_REACHED'] = "RetryLimitReached";
        StatusEnum[StatusEnum["Unchecked"] = 'UNCHECKED'] = "Unchecked";
    })(StatusEnum = KYCCheckStatusData.StatusEnum || (KYCCheckStatusData.StatusEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["BankAccountVerification"] = 'BANK_ACCOUNT_VERIFICATION'] = "BankAccountVerification";
        TypeEnum[TypeEnum["CardVerification"] = 'CARD_VERIFICATION'] = "CardVerification";
        TypeEnum[TypeEnum["CompanyVerification"] = 'COMPANY_VERIFICATION'] = "CompanyVerification";
        TypeEnum[TypeEnum["IdentityVerification"] = 'IDENTITY_VERIFICATION'] = "IdentityVerification";
        TypeEnum[TypeEnum["LegalArrangementVerification"] = 'LEGAL_ARRANGEMENT_VERIFICATION'] = "LegalArrangementVerification";
        TypeEnum[TypeEnum["NonprofitVerification"] = 'NONPROFIT_VERIFICATION'] = "NonprofitVerification";
        TypeEnum[TypeEnum["PassportVerification"] = 'PASSPORT_VERIFICATION'] = "PassportVerification";
        TypeEnum[TypeEnum["PayoutMethodVerification"] = 'PAYOUT_METHOD_VERIFICATION'] = "PayoutMethodVerification";
        TypeEnum[TypeEnum["PciVerification"] = 'PCI_VERIFICATION'] = "PciVerification";
    })(TypeEnum = KYCCheckStatusData.TypeEnum || (KYCCheckStatusData.TypeEnum = {}));
})(KYCCheckStatusData = exports.KYCCheckStatusData || (exports.KYCCheckStatusData = {}));
