// @adyen/api-library/lib/src/typings/platformsAccount/legalArrangementDetail.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LegalArrangementDetail = void 0;
class LegalArrangementDetail {
    static getAttributeTypeMap() {
        return LegalArrangementDetail.attributeTypeMap;
    }
}
exports.LegalArrangementDetail = LegalArrangementDetail;
LegalArrangementDetail.discriminator = undefined;
LegalArrangementDetail.attributeTypeMap = [
    {
        "name": "address",
        "baseName": "address",
        "type": "ViasAddress"
    },
    {
        "name": "legalArrangementCode",
        "baseName": "legalArrangementCode",
        "type": "string"
    },
    {
        "name": "legalArrangementEntities",
        "baseName": "legalArrangementEntities",
        "type": "Array<LegalArrangementEntityDetail>"
    },
    {
        "name": "legalArrangementReference",
        "baseName": "legalArrangementReference",
        "type": "string"
    },
    {
        "name": "legalForm",
        "baseName": "legalForm",
        "type": "LegalArrangementDetail.LegalFormEnum"
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "registrationNumber",
        "baseName": "registrationNumber",
        "type": "string"
    },
    {
        "name": "taxNumber",
        "baseName": "taxNumber",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "LegalArrangementDetail.TypeEnum"
    }
];
(function (LegalArrangementDetail) {
    let LegalFormEnum;
    (function (LegalFormEnum) {
        LegalFormEnum[LegalFormEnum["CashManagementTrust"] = 'CashManagementTrust'] = "CashManagementTrust";
        LegalFormEnum[LegalFormEnum["CorporateUnitTrust"] = 'CorporateUnitTrust'] = "CorporateUnitTrust";
        LegalFormEnum[LegalFormEnum["DeceasedEstate"] = 'DeceasedEstate'] = "DeceasedEstate";
        LegalFormEnum[LegalFormEnum["DiscretionaryInvestmentTrust"] = 'DiscretionaryInvestmentTrust'] = "DiscretionaryInvestmentTrust";
        LegalFormEnum[LegalFormEnum["DiscretionaryServicesManagementTrust"] = 'DiscretionaryServicesManagementTrust'] = "DiscretionaryServicesManagementTrust";
        LegalFormEnum[LegalFormEnum["DiscretionaryTradingTrust"] = 'DiscretionaryTradingTrust'] = "DiscretionaryTradingTrust";
        LegalFormEnum[LegalFormEnum["FirstHomeSaverAccountsTrust"] = 'FirstHomeSaverAccountsTrust'] = "FirstHomeSaverAccountsTrust";
        LegalFormEnum[LegalFormEnum["FixedTrust"] = 'FixedTrust'] = "FixedTrust";
        LegalFormEnum[LegalFormEnum["FixedUnitTrust"] = 'FixedUnitTrust'] = "FixedUnitTrust";
        LegalFormEnum[LegalFormEnum["HybridTrust"] = 'HybridTrust'] = "HybridTrust";
        LegalFormEnum[LegalFormEnum["ListedPublicUnitTrust"] = 'ListedPublicUnitTrust'] = "ListedPublicUnitTrust";
        LegalFormEnum[LegalFormEnum["OtherTrust"] = 'OtherTrust'] = "OtherTrust";
        LegalFormEnum[LegalFormEnum["PooledSuperannuationTrust"] = 'PooledSuperannuationTrust'] = "PooledSuperannuationTrust";
        LegalFormEnum[LegalFormEnum["PublicTradingTrust"] = 'PublicTradingTrust'] = "PublicTradingTrust";
        LegalFormEnum[LegalFormEnum["UnlistedPublicUnitTrust"] = 'UnlistedPublicUnitTrust'] = "UnlistedPublicUnitTrust";
        LegalFormEnum[LegalFormEnum["LimitedPartnership"] = 'LimitedPartnership'] = "LimitedPartnership";
        LegalFormEnum[LegalFormEnum["FamilyPartnership"] = 'FamilyPartnership'] = "FamilyPartnership";
        LegalFormEnum[LegalFormEnum["OtherPartnership"] = 'OtherPartnership'] = "OtherPartnership";
    })(LegalFormEnum = LegalArrangementDetail.LegalFormEnum || (LegalArrangementDetail.LegalFormEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["Association"] = 'Association'] = "Association";
        TypeEnum[TypeEnum["Partnership"] = 'Partnership'] = "Partnership";
        TypeEnum[TypeEnum["SoleProprietorship"] = 'SoleProprietorship'] = "SoleProprietorship";
        TypeEnum[TypeEnum["Trust"] = 'Trust'] = "Trust";
    })(TypeEnum = LegalArrangementDetail.TypeEnum || (LegalArrangementDetail.TypeEnum = {}));
})(LegalArrangementDetail = exports.LegalArrangementDetail || (exports.LegalArrangementDetail = {}));
