// @adyen/api-library/lib/src/typings/platformsAccount/legalArrangementEntityDetail.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LegalArrangementEntityDetail = void 0;
class LegalArrangementEntityDetail {
    static getAttributeTypeMap() {
        return LegalArrangementEntityDetail.attributeTypeMap;
    }
}
exports.LegalArrangementEntityDetail = LegalArrangementEntityDetail;
LegalArrangementEntityDetail.discriminator = undefined;
LegalArrangementEntityDetail.attributeTypeMap = [
    {
        "name": "address",
        "baseName": "address",
        "type": "ViasAddress"
    },
    {
        "name": "businessDetails",
        "baseName": "businessDetails",
        "type": "BusinessDetails"
    },
    {
        "name": "email",
        "baseName": "email",
        "type": "string"
    },
    {
        "name": "fullPhoneNumber",
        "baseName": "fullPhoneNumber",
        "type": "string"
    },
    {
        "name": "individualDetails",
        "baseName": "individualDetails",
        "type": "IndividualDetails"
    },
    {
        "name": "legalArrangementEntityCode",
        "baseName": "legalArrangementEntityCode",
        "type": "string"
    },
    {
        "name": "legalArrangementEntityReference",
        "baseName": "legalArrangementEntityReference",
        "type": "string"
    },
    {
        "name": "legalArrangementMembers",
        "baseName": "legalArrangementMembers",
        "type": "Array<LegalArrangementEntityDetail.LegalArrangementMembersEnum>"
    },
    {
        "name": "legalEntityType",
        "baseName": "legalEntityType",
        "type": "LegalArrangementEntityDetail.LegalEntityTypeEnum"
    },
    {
        "name": "phoneNumber",
        "baseName": "phoneNumber",
        "type": "ViasPhoneNumber"
    },
    {
        "name": "webAddress",
        "baseName": "webAddress",
        "type": "string"
    }
];
(function (LegalArrangementEntityDetail) {
    let LegalArrangementMembersEnum;
    (function (LegalArrangementMembersEnum) {
        LegalArrangementMembersEnum[LegalArrangementMembersEnum["Beneficiary"] = 'Beneficiary'] = "Beneficiary";
        LegalArrangementMembersEnum[LegalArrangementMembersEnum["ControllingPerson"] = 'ControllingPerson'] = "ControllingPerson";
        LegalArrangementMembersEnum[LegalArrangementMembersEnum["Partner"] = 'Partner'] = "Partner";
        LegalArrangementMembersEnum[LegalArrangementMembersEnum["Protector"] = 'Protector'] = "Protector";
        LegalArrangementMembersEnum[LegalArrangementMembersEnum["Settlor"] = 'Settlor'] = "Settlor";
        LegalArrangementMembersEnum[LegalArrangementMembersEnum["Shareholder"] = 'Shareholder'] = "Shareholder";
        LegalArrangementMembersEnum[LegalArrangementMembersEnum["Trustee"] = 'Trustee'] = "Trustee";
    })(LegalArrangementMembersEnum = LegalArrangementEntityDetail.LegalArrangementMembersEnum || (LegalArrangementEntityDetail.LegalArrangementMembersEnum = {}));
    let LegalEntityTypeEnum;
    (function (LegalEntityTypeEnum) {
        LegalEntityTypeEnum[LegalEntityTypeEnum["Business"] = 'Business'] = "Business";
        LegalEntityTypeEnum[LegalEntityTypeEnum["Individual"] = 'Individual'] = "Individual";
        LegalEntityTypeEnum[LegalEntityTypeEnum["NonProfit"] = 'NonProfit'] = "NonProfit";
        LegalEntityTypeEnum[LegalEntityTypeEnum["Partnership"] = 'Partnership'] = "Partnership";
        LegalEntityTypeEnum[LegalEntityTypeEnum["PublicCompany"] = 'PublicCompany'] = "PublicCompany";
    })(LegalEntityTypeEnum = LegalArrangementEntityDetail.LegalEntityTypeEnum || (LegalArrangementEntityDetail.LegalEntityTypeEnum = {}));
})(LegalArrangementEntityDetail = exports.LegalArrangementEntityDetail || (exports.LegalArrangementEntityDetail = {}));
