// @adyen/api-library/lib/src/typings/platformsAccount/storeDetail.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StoreDetail = void 0;
class StoreDetail {
    static getAttributeTypeMap() {
        return StoreDetail.attributeTypeMap;
    }
}
exports.StoreDetail = StoreDetail;
StoreDetail.discriminator = undefined;
StoreDetail.attributeTypeMap = [
    {
        "name": "address",
        "baseName": "address",
        "type": "ViasAddress"
    },
    {
        "name": "fullPhoneNumber",
        "baseName": "fullPhoneNumber",
        "type": "string"
    },
    {
        "name": "merchantAccount",
        "baseName": "merchantAccount",
        "type": "string"
    },
    {
        "name": "merchantCategoryCode",
        "baseName": "merchantCategoryCode",
        "type": "string"
    },
    {
        "name": "phoneNumber",
        "baseName": "phoneNumber",
        "type": "ViasPhoneNumber"
    },
    {
        "name": "shopperInteraction",
        "baseName": "shopperInteraction",
        "type": "StoreDetail.ShopperInteractionEnum"
    },
    {
        "name": "splitConfigurationUUID",
        "baseName": "splitConfigurationUUID",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "StoreDetail.StatusEnum"
    },
    {
        "name": "store",
        "baseName": "store",
        "type": "string"
    },
    {
        "name": "storeName",
        "baseName": "storeName",
        "type": "string"
    },
    {
        "name": "storeReference",
        "baseName": "storeReference",
        "type": "string"
    },
    {
        "name": "virtualAccount",
        "baseName": "virtualAccount",
        "type": "string"
    },
    {
        "name": "webAddress",
        "baseName": "webAddress",
        "type": "string"
    }
];
(function (StoreDetail) {
    let ShopperInteractionEnum;
    (function (ShopperInteractionEnum) {
        ShopperInteractionEnum[ShopperInteractionEnum["Ecommerce"] = 'Ecommerce'] = "Ecommerce";
        ShopperInteractionEnum[ShopperInteractionEnum["Pos"] = 'POS'] = "Pos";
    })(ShopperInteractionEnum = StoreDetail.ShopperInteractionEnum || (StoreDetail.ShopperInteractionEnum = {}));
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["Active"] = 'Active'] = "Active";
        StatusEnum[StatusEnum["Closed"] = 'Closed'] = "Closed";
        StatusEnum[StatusEnum["Inactive"] = 'Inactive'] = "Inactive";
        StatusEnum[StatusEnum["InactiveWithModifications"] = 'InactiveWithModifications'] = "InactiveWithModifications";
        StatusEnum[StatusEnum["Pending"] = 'Pending'] = "Pending";
    })(StatusEnum = StoreDetail.StatusEnum || (StoreDetail.StatusEnum = {}));
})(StoreDetail = exports.StoreDetail || (exports.StoreDetail = {}));
