// @adyen/api-library/lib/src/typings/platformsAccount/updatePayoutScheduleRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdatePayoutScheduleRequest = void 0;
class UpdatePayoutScheduleRequest {
    static getAttributeTypeMap() {
        return UpdatePayoutScheduleRequest.attributeTypeMap;
    }
}
exports.UpdatePayoutScheduleRequest = UpdatePayoutScheduleRequest;
UpdatePayoutScheduleRequest.discriminator = undefined;
UpdatePayoutScheduleRequest.attributeTypeMap = [
    {
        "name": "action",
        "baseName": "action",
        "type": "UpdatePayoutScheduleRequest.ActionEnum"
    },
    {
        "name": "reason",
        "baseName": "reason",
        "type": "string"
    },
    {
        "name": "schedule",
        "baseName": "schedule",
        "type": "UpdatePayoutScheduleRequest.ScheduleEnum"
    }
];
(function (UpdatePayoutScheduleRequest) {
    let ActionEnum;
    (function (ActionEnum) {
        ActionEnum[ActionEnum["Close"] = 'CLOSE'] = "Close";
        ActionEnum[ActionEnum["Nothing"] = 'NOTHING'] = "Nothing";
        ActionEnum[ActionEnum["Update"] = 'UPDATE'] = "Update";
    })(ActionEnum = UpdatePayoutScheduleRequest.ActionEnum || (UpdatePayoutScheduleRequest.ActionEnum = {}));
    let ScheduleEnum;
    (function (ScheduleEnum) {
        ScheduleEnum[ScheduleEnum["BiweeklyOn1StAnd15ThAtMidnight"] = 'BIWEEKLY_ON_1ST_AND_15TH_AT_MIDNIGHT'] = "BiweeklyOn1StAnd15ThAtMidnight";
        ScheduleEnum[ScheduleEnum["BiweeklyOn1StAnd15ThAtNoon"] = 'BIWEEKLY_ON_1ST_AND_15TH_AT_NOON'] = "BiweeklyOn1StAnd15ThAtNoon";
        ScheduleEnum[ScheduleEnum["BiDailyAu"] = 'BI_DAILY_AU'] = "BiDailyAu";
        ScheduleEnum[ScheduleEnum["BiDailyEu"] = 'BI_DAILY_EU'] = "BiDailyEu";
        ScheduleEnum[ScheduleEnum["BiDailyUs"] = 'BI_DAILY_US'] = "BiDailyUs";
        ScheduleEnum[ScheduleEnum["Daily"] = 'DAILY'] = "Daily";
        ScheduleEnum[ScheduleEnum["Daily6Pm"] = 'DAILY_6PM'] = "Daily6Pm";
        ScheduleEnum[ScheduleEnum["DailyAu"] = 'DAILY_AU'] = "DailyAu";
        ScheduleEnum[ScheduleEnum["DailyEu"] = 'DAILY_EU'] = "DailyEu";
        ScheduleEnum[ScheduleEnum["DailySg"] = 'DAILY_SG'] = "DailySg";
        ScheduleEnum[ScheduleEnum["DailyUs"] = 'DAILY_US'] = "DailyUs";
        ScheduleEnum[ScheduleEnum["Default"] = 'DEFAULT'] = "Default";
        ScheduleEnum[ScheduleEnum["Every6HoursFromMidnight"] = 'EVERY_6_HOURS_FROM_MIDNIGHT'] = "Every6HoursFromMidnight";
        ScheduleEnum[ScheduleEnum["Hold"] = 'HOLD'] = "Hold";
        ScheduleEnum[ScheduleEnum["Monthly"] = 'MONTHLY'] = "Monthly";
        ScheduleEnum[ScheduleEnum["MonthlyOn15ThAtMidnight"] = 'MONTHLY_ON_15TH_AT_MIDNIGHT'] = "MonthlyOn15ThAtMidnight";
        ScheduleEnum[ScheduleEnum["Weekly"] = 'WEEKLY'] = "Weekly";
        ScheduleEnum[ScheduleEnum["WeeklyMonToFriAu"] = 'WEEKLY_MON_TO_FRI_AU'] = "WeeklyMonToFriAu";
        ScheduleEnum[ScheduleEnum["WeeklyMonToFriEu"] = 'WEEKLY_MON_TO_FRI_EU'] = "WeeklyMonToFriEu";
        ScheduleEnum[ScheduleEnum["WeeklyMonToFriUs"] = 'WEEKLY_MON_TO_FRI_US'] = "WeeklyMonToFriUs";
        ScheduleEnum[ScheduleEnum["WeeklyOnTueFriMidnight"] = 'WEEKLY_ON_TUE_FRI_MIDNIGHT'] = "WeeklyOnTueFriMidnight";
        ScheduleEnum[ScheduleEnum["WeeklySunToThuAu"] = 'WEEKLY_SUN_TO_THU_AU'] = "WeeklySunToThuAu";
        ScheduleEnum[ScheduleEnum["WeeklySunToThuUs"] = 'WEEKLY_SUN_TO_THU_US'] = "WeeklySunToThuUs";
        ScheduleEnum[ScheduleEnum["Yearly"] = 'YEARLY'] = "Yearly";
    })(ScheduleEnum = UpdatePayoutScheduleRequest.ScheduleEnum || (UpdatePayoutScheduleRequest.ScheduleEnum = {}));
})(UpdatePayoutScheduleRequest = exports.UpdatePayoutScheduleRequest || (exports.UpdatePayoutScheduleRequest = {}));
