// @adyen/api-library/lib/src/typings/platformsAccount/updateAccountHolderRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateAccountHolderRequest = void 0;
class UpdateAccountHolderRequest {
    static getAttributeTypeMap() {
        return UpdateAccountHolderRequest.attributeTypeMap;
    }
}
exports.UpdateAccountHolderRequest = UpdateAccountHolderRequest;
UpdateAccountHolderRequest.discriminator = undefined;
UpdateAccountHolderRequest.attributeTypeMap = [
    {
        "name": "accountHolderCode",
        "baseName": "accountHolderCode",
        "type": "string"
    },
    {
        "name": "accountHolderDetails",
        "baseName": "accountHolderDetails",
        "type": "AccountHolderDetails"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "legalEntity",
        "baseName": "legalEntity",
        "type": "UpdateAccountHolderRequest.LegalEntityEnum"
    },
    {
        "name": "primaryCurrency",
        "baseName": "primaryCurrency",
        "type": "string"
    },
    {
        "name": "processingTier",
        "baseName": "processingTier",
        "type": "number"
    },
    {
        "name": "verificationProfile",
        "baseName": "verificationProfile",
        "type": "string"
    }
];
(function (UpdateAccountHolderRequest) {
    let LegalEntityEnum;
    (function (LegalEntityEnum) {
        LegalEntityEnum[LegalEntityEnum["Business"] = 'Business'] = "Business";
        LegalEntityEnum[LegalEntityEnum["Individual"] = 'Individual'] = "Individual";
        LegalEntityEnum[LegalEntityEnum["NonProfit"] = 'NonProfit'] = "NonProfit";
        LegalEntityEnum[LegalEntityEnum["Partnership"] = 'Partnership'] = "Partnership";
        LegalEntityEnum[LegalEntityEnum["PublicCompany"] = 'PublicCompany'] = "PublicCompany";
    })(LegalEntityEnum = UpdateAccountHolderRequest.LegalEntityEnum || (UpdateAccountHolderRequest.LegalEntityEnum = {}));
})(UpdateAccountHolderRequest = exports.UpdateAccountHolderRequest || (exports.UpdateAccountHolderRequest = {}));
